// dana-node/dist/merchant_management/v1/models/BusinessDocs.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessDocsDocTypeEnum = void 0;
exports.instanceOfBusinessDocs = instanceOfBusinessDocs;
exports.BusinessDocsFromJSON = BusinessDocsFromJSON;
exports.BusinessDocsFromJSONTyped = BusinessDocsFromJSONTyped;
exports.BusinessDocsToJSON = BusinessDocsToJSON;
exports.BusinessDocsToJSONTyped = BusinessDocsToJSONTyped;
exports.validateBusinessDocs = validateBusinessDocs;
/**
 * @export
 */
exports.BusinessDocsDocTypeEnum = {
    Ktp: 'KTP',
    Sim: 'SIM',
    Siup: 'SIUP',
    Nib: 'NIB'
};
/**
 * Check if a given object implements the BusinessDocs interface.
 */
function instanceOfBusinessDocs(value) {
    return true;
}
function BusinessDocsFromJSON(json) {
    return BusinessDocsFromJSONTyped(json, false);
}
function BusinessDocsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'docType': json['docType'] == null ? undefined : json['docType'],
        'docId': json['docId'] == null ? undefined : json['docId'],
        'docFile': json['docFile'] == null ? undefined : json['docFile'],
    };
}
function BusinessDocsToJSON(json) {
    return BusinessDocsToJSONTyped(json, false);
}
function BusinessDocsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'docType': value['docType'],
        'docId': value['docId'],
        'docFile': value['docFile'],
    };
}
const propertyValidationAttributesMap = {};
function validateBusinessDocs(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
