// dana-node/dist/merchant_management/v1/models/CreateShopResponseResponseHead.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateShopResponseResponseHeadFunctionEnum = void 0;
exports.instanceOfCreateShopResponseResponseHead = instanceOfCreateShopResponseResponseHead;
exports.CreateShopResponseResponseHeadFromJSON = CreateShopResponseResponseHeadFromJSON;
exports.CreateShopResponseResponseHeadFromJSONTyped = CreateShopResponseResponseHeadFromJSONTyped;
exports.CreateShopResponseResponseHeadToJSON = CreateShopResponseResponseHeadToJSON;
exports.CreateShopResponseResponseHeadToJSONTyped = CreateShopResponseResponseHeadToJSONTyped;
exports.validateCreateShopResponseResponseHead = validateCreateShopResponseResponseHead;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.CreateShopResponseResponseHeadFunctionEnum = {
    DanaMerchantShopCreateShop: 'dana.merchant.shop.createShop'
};
/**
 * Check if a given object implements the CreateShopResponseResponseHead interface.
 */
function instanceOfCreateShopResponseResponseHead(value) {
    return true;
}
function CreateShopResponseResponseHeadFromJSON(json) {
    return CreateShopResponseResponseHeadFromJSONTyped(json, false);
}
function CreateShopResponseResponseHeadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'version': json['version'] == null ? undefined : json['version'],
        '_function': json['function'] == null ? undefined : json['function'],
        'clientId': json['clientId'] == null ? undefined : json['clientId'],
        'respTime': json['respTime'] == null ? undefined : json['respTime'],
        'reqMsgId': json['reqMsgId'] == null ? undefined : json['reqMsgId'],
    };
}
function CreateShopResponseResponseHeadToJSON(json) {
    return CreateShopResponseResponseHeadToJSONTyped(json, false);
}
function CreateShopResponseResponseHeadToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'version': value['version'],
        'function': value['_function'],
        'clientId': value['clientId'],
        'respTime': value['respTime'],
        'reqMsgId': value['reqMsgId'],
    };
}
const propertyValidationAttributesMap = {
    version: {
        maxLength: 8,
    },
    _function: {
        maxLength: 128,
    },
    clientId: {
        maxLength: 32,
    },
    reqMsgId: {
        maxLength: 64,
    },
};
function validateCreateShopResponseResponseHead(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('version', value.version, propertyValidationAttributesMap['version']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('_function', value._function, propertyValidationAttributesMap['_function']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientId', value.clientId, propertyValidationAttributesMap['clientId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reqMsgId', value.reqMsgId, propertyValidationAttributesMap['reqMsgId']));
    return validationErrorContexts;
}
