// dana-node/dist/merchant_management/v1/models/CreateDivisionResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateDivisionResponse = instanceOfCreateDivisionResponse;
exports.CreateDivisionResponseFromJSON = CreateDivisionResponseFromJSON;
exports.CreateDivisionResponseFromJSONTyped = CreateDivisionResponseFromJSONTyped;
exports.CreateDivisionResponseToJSON = CreateDivisionResponseToJSON;
exports.CreateDivisionResponseToJSONTyped = CreateDivisionResponseToJSONTyped;
exports.validateCreateDivisionResponse = validateCreateDivisionResponse;
const CreateDivisionResponseResponse_1 = require("./CreateDivisionResponseResponse");
/**
 * Check if a given object implements the CreateDivisionResponse interface.
 */
function instanceOfCreateDivisionResponse(value) {
    if (!('response' in value) || value['response'] === undefined)
        return false;
    return true;
}
function CreateDivisionResponseFromJSON(json) {
    return CreateDivisionResponseFromJSONTyped(json, false);
}
function CreateDivisionResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'response': (0, CreateDivisionResponseResponse_1.CreateDivisionResponseResponseFromJSON)(json['response']),
        'signature': json['signature'] == null ? undefined : json['signature'],
    };
}
function CreateDivisionResponseToJSON(json) {
    return CreateDivisionResponseToJSONTyped(json, false);
}
function CreateDivisionResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'response': (0, CreateDivisionResponseResponse_1.CreateDivisionResponseResponseToJSON)(value['response']),
        'signature': value['signature'],
    };
}
const propertyValidationAttributesMap = {};
function validateCreateDivisionResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, CreateDivisionResponseResponse_1.validateCreateDivisionResponseResponse)(value.response));
    return validationErrorContexts;
}
