// dana-node/dist/merchant_management/v1/models/CreateDivisionRequestExtInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDivisionRequestExtInfoUSERPROFILINGEnum = exports.CreateDivisionRequestExtInfoGOODSSOLDTYPEEnum = void 0;
exports.instanceOfCreateDivisionRequestExtInfo = instanceOfCreateDivisionRequestExtInfo;
exports.CreateDivisionRequestExtInfoFromJSON = CreateDivisionRequestExtInfoFromJSON;
exports.CreateDivisionRequestExtInfoFromJSONTyped = CreateDivisionRequestExtInfoFromJSONTyped;
exports.CreateDivisionRequestExtInfoToJSON = CreateDivisionRequestExtInfoToJSON;
exports.CreateDivisionRequestExtInfoToJSONTyped = CreateDivisionRequestExtInfoToJSONTyped;
exports.validateCreateDivisionRequestExtInfo = validateCreateDivisionRequestExtInfo;
/**
 * @export
 */
exports.CreateDivisionRequestExtInfoGOODSSOLDTYPEEnum = {
    Digital: 'DIGITAL',
    Physical: 'PHYSICAL'
};
/**
 * @export
 */
exports.CreateDivisionRequestExtInfoUSERPROFILINGEnum = {
    B2B: 'B2B',
    B2C: 'B2C'
};
/**
 * Check if a given object implements the CreateDivisionRequestExtInfo interface.
 */
function instanceOfCreateDivisionRequestExtInfo(value) {
    return true;
}
function CreateDivisionRequestExtInfoFromJSON(json) {
    return CreateDivisionRequestExtInfoFromJSONTyped(json, false);
}
function CreateDivisionRequestExtInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'pICEMAIL': json['PIC_EMAIL'] == null ? undefined : json['PIC_EMAIL'],
        'pICPHONENUMBER': json['PIC_PHONENUMBER'] == null ? undefined : json['PIC_PHONENUMBER'],
        'sUBMITTEREMAIL': json['SUBMITTER_EMAIL'] == null ? undefined : json['SUBMITTER_EMAIL'],
        'gOODSSOLDTYPE': json['GOODS_SOLD_TYPE'] == null ? undefined : json['GOODS_SOLD_TYPE'],
        'uSECASE': json['USECASE'] == null ? undefined : json['USECASE'],
        'uSERPROFILING': json['USER_PROFILING'] == null ? undefined : json['USER_PROFILING'],
        'aVGTICKET': json['AVG_TICKET'] == null ? undefined : json['AVG_TICKET'],
        'oMZET': json['OMZET'] == null ? undefined : json['OMZET'],
        'eXTURLS': json['EXT_URLS'] == null ? undefined : json['EXT_URLS'],
        'bRANDNAME': json['BRAND_NAME'] == null ? undefined : json['BRAND_NAME'],
    };
}
function CreateDivisionRequestExtInfoToJSON(json) {
    return CreateDivisionRequestExtInfoToJSONTyped(json, false);
}
function CreateDivisionRequestExtInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'PIC_EMAIL': value['pICEMAIL'],
        'PIC_PHONENUMBER': value['pICPHONENUMBER'],
        'SUBMITTER_EMAIL': value['sUBMITTEREMAIL'],
        'GOODS_SOLD_TYPE': value['gOODSSOLDTYPE'],
        'USECASE': value['uSECASE'],
        'USER_PROFILING': value['uSERPROFILING'],
        'AVG_TICKET': value['aVGTICKET'],
        'OMZET': value['oMZET'],
        'EXT_URLS': value['eXTURLS'],
        'BRAND_NAME': value['bRANDNAME'],
    };
}
const propertyValidationAttributesMap = {};
function validateCreateDivisionRequestExtInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
