// dana-node/dist/merchant_management/v1/models/CreateDivisionResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateDivisionResponseResponse = instanceOfCreateDivisionResponseResponse;
exports.CreateDivisionResponseResponseFromJSON = CreateDivisionResponseResponseFromJSON;
exports.CreateDivisionResponseResponseFromJSONTyped = CreateDivisionResponseResponseFromJSONTyped;
exports.CreateDivisionResponseResponseToJSON = CreateDivisionResponseResponseToJSON;
exports.CreateDivisionResponseResponseToJSONTyped = CreateDivisionResponseResponseToJSONTyped;
exports.validateCreateDivisionResponseResponse = validateCreateDivisionResponseResponse;
const CreateDivisionResponseResponseBody_1 = require("./CreateDivisionResponseResponseBody");
const CreateDivisionResponseResponseHead_1 = require("./CreateDivisionResponseResponseHead");
/**
 * Check if a given object implements the CreateDivisionResponseResponse interface.
 */
function instanceOfCreateDivisionResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function CreateDivisionResponseResponseFromJSON(json) {
    return CreateDivisionResponseResponseFromJSONTyped(json, false);
}
function CreateDivisionResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, CreateDivisionResponseResponseHead_1.CreateDivisionResponseResponseHeadFromJSON)(json['head']),
        'body': (0, CreateDivisionResponseResponseBody_1.CreateDivisionResponseResponseBodyFromJSON)(json['body']),
    };
}
function CreateDivisionResponseResponseToJSON(json) {
    return CreateDivisionResponseResponseToJSONTyped(json, false);
}
function CreateDivisionResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, CreateDivisionResponseResponseHead_1.CreateDivisionResponseResponseHeadToJSON)(value['head']),
        'body': (0, CreateDivisionResponseResponseBody_1.CreateDivisionResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateCreateDivisionResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, CreateDivisionResponseResponseHead_1.validateCreateDivisionResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, CreateDivisionResponseResponseBody_1.validateCreateDivisionResponseResponseBody)(value.body));
    return validationErrorContexts;
}
