// dana-node/dist/merchant_management/v1/models/CreateDivisionResponseResponseHead.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateDivisionResponseResponseHeadFunctionEnum = void 0;
exports.instanceOfCreateDivisionResponseResponseHead = instanceOfCreateDivisionResponseResponseHead;
exports.CreateDivisionResponseResponseHeadFromJSON = CreateDivisionResponseResponseHeadFromJSON;
exports.CreateDivisionResponseResponseHeadFromJSONTyped = CreateDivisionResponseResponseHeadFromJSONTyped;
exports.CreateDivisionResponseResponseHeadToJSON = CreateDivisionResponseResponseHeadToJSON;
exports.CreateDivisionResponseResponseHeadToJSONTyped = CreateDivisionResponseResponseHeadToJSONTyped;
exports.validateCreateDivisionResponseResponseHead = validateCreateDivisionResponseResponseHead;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.CreateDivisionResponseResponseHeadFunctionEnum = {
    DanaMerchantDivisionCreateDivision: 'dana.merchant.division.createDivision'
};
/**
 * Check if a given object implements the CreateDivisionResponseResponseHead interface.
 */
function instanceOfCreateDivisionResponseResponseHead(value) {
    return true;
}
function CreateDivisionResponseResponseHeadFromJSON(json) {
    return CreateDivisionResponseResponseHeadFromJSONTyped(json, false);
}
function CreateDivisionResponseResponseHeadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'version': json['version'] == null ? undefined : json['version'],
        '_function': json['function'] == null ? undefined : json['function'],
        'clientId': json['clientId'] == null ? undefined : json['clientId'],
        'clientSecret': json['clientSecret'] == null ? undefined : json['clientSecret'],
        'respTime': json['respTime'] == null ? undefined : json['respTime'],
        'reqMsgId': json['reqMsgId'] == null ? undefined : json['reqMsgId'],
        'reserve': json['reserve'] == null ? undefined : json['reserve'],
    };
}
function CreateDivisionResponseResponseHeadToJSON(json) {
    return CreateDivisionResponseResponseHeadToJSONTyped(json, false);
}
function CreateDivisionResponseResponseHeadToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'version': value['version'],
        'function': value['_function'],
        'clientId': value['clientId'],
        'clientSecret': value['clientSecret'],
        'respTime': value['respTime'],
        'reqMsgId': value['reqMsgId'],
        'reserve': value['reserve'],
    };
}
const propertyValidationAttributesMap = {
    version: {
        maxLength: 8,
    },
    _function: {
        maxLength: 128,
    },
    clientId: {
        maxLength: 36,
    },
    clientSecret: {
        maxLength: 64,
    },
    respTime: {
        maxLength: 25,
    },
    reqMsgId: {
        maxLength: 64,
    },
    reserve: {
        maxLength: 256,
    },
};
function validateCreateDivisionResponseResponseHead(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('version', value.version, propertyValidationAttributesMap['version']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('_function', value._function, propertyValidationAttributesMap['_function']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientId', value.clientId, propertyValidationAttributesMap['clientId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientSecret', value.clientSecret, propertyValidationAttributesMap['clientSecret']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('respTime', value.respTime, propertyValidationAttributesMap['respTime']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reqMsgId', value.reqMsgId, propertyValidationAttributesMap['reqMsgId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reserve', value.reserve, propertyValidationAttributesMap['reserve']));
    return validationErrorContexts;
}
