// dana-node/dist/merchant_management/v1/models/CreateShopResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateShopResponseResponse = instanceOfCreateShopResponseResponse;
exports.CreateShopResponseResponseFromJSON = CreateShopResponseResponseFromJSON;
exports.CreateShopResponseResponseFromJSONTyped = CreateShopResponseResponseFromJSONTyped;
exports.CreateShopResponseResponseToJSON = CreateShopResponseResponseToJSON;
exports.CreateShopResponseResponseToJSONTyped = CreateShopResponseResponseToJSONTyped;
exports.validateCreateShopResponseResponse = validateCreateShopResponseResponse;
const CreateShopResponseResponseHead_1 = require("./CreateShopResponseResponseHead");
const CreateShopResponseResponseBody_1 = require("./CreateShopResponseResponseBody");
/**
 * Check if a given object implements the CreateShopResponseResponse interface.
 */
function instanceOfCreateShopResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function CreateShopResponseResponseFromJSON(json) {
    return CreateShopResponseResponseFromJSONTyped(json, false);
}
function CreateShopResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, CreateShopResponseResponseHead_1.CreateShopResponseResponseHeadFromJSON)(json['head']),
        'body': (0, CreateShopResponseResponseBody_1.CreateShopResponseResponseBodyFromJSON)(json['body']),
    };
}
function CreateShopResponseResponseToJSON(json) {
    return CreateShopResponseResponseToJSONTyped(json, false);
}
function CreateShopResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, CreateShopResponseResponseHead_1.CreateShopResponseResponseHeadToJSON)(value['head']),
        'body': (0, CreateShopResponseResponseBody_1.CreateShopResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateCreateShopResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, CreateShopResponseResponseHead_1.validateCreateShopResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, CreateShopResponseResponseBody_1.validateCreateShopResponseResponseBody)(value.body));
    return validationErrorContexts;
}
