// dana-node/dist/merchant_management/v1/models/CreateShopResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateShopResponse = instanceOfCreateShopResponse;
exports.CreateShopResponseFromJSON = CreateShopResponseFromJSON;
exports.CreateShopResponseFromJSONTyped = CreateShopResponseFromJSONTyped;
exports.CreateShopResponseToJSON = CreateShopResponseToJSON;
exports.CreateShopResponseToJSONTyped = CreateShopResponseToJSONTyped;
exports.validateCreateShopResponse = validateCreateShopResponse;
const CreateShopResponseResponse_1 = require("./CreateShopResponseResponse");
/**
 * Check if a given object implements the CreateShopResponse interface.
 */
function instanceOfCreateShopResponse(value) {
    if (!('response' in value) || value['response'] === undefined)
        return false;
    return true;
}
function CreateShopResponseFromJSON(json) {
    return CreateShopResponseFromJSONTyped(json, false);
}
function CreateShopResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'response': (0, CreateShopResponseResponse_1.CreateShopResponseResponseFromJSON)(json['response']),
    };
}
function CreateShopResponseToJSON(json) {
    return CreateShopResponseToJSONTyped(json, false);
}
function CreateShopResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'response': (0, CreateShopResponseResponse_1.CreateShopResponseResponseToJSON)(value['response']),
    };
}
const propertyValidationAttributesMap = {};
function validateCreateShopResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, CreateShopResponseResponse_1.validateCreateShopResponseResponse)(value.response));
    return validationErrorContexts;
}
