// dana-node/dist/merchant_management/v1/models/MerchantResourceInformation.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerchantResourceInformationResourceTypeEnum = void 0;
exports.instanceOfMerchantResourceInformation = instanceOfMerchantResourceInformation;
exports.MerchantResourceInformationFromJSON = MerchantResourceInformationFromJSON;
exports.MerchantResourceInformationFromJSONTyped = MerchantResourceInformationFromJSONTyped;
exports.MerchantResourceInformationToJSON = MerchantResourceInformationToJSON;
exports.MerchantResourceInformationToJSONTyped = MerchantResourceInformationToJSONTyped;
exports.validateMerchantResourceInformation = validateMerchantResourceInformation;
/**
 * @export
 */
exports.MerchantResourceInformationResourceTypeEnum = {
    MerchantDepositBalance: 'MERCHANT_DEPOSIT_BALANCE',
    MerchantAvailableBalance: 'MERCHANT_AVAILABLE_BALANCE',
    MerchantTotalBalance: 'MERCHANT_TOTAL_BALANCE'
};
/**
 * Check if a given object implements the MerchantResourceInformation interface.
 */
function instanceOfMerchantResourceInformation(value) {
    if (!('resourceType' in value) || value['resourceType'] === undefined)
        return false;
    if (!('value' in value) || value['value'] === undefined)
        return false;
    return true;
}
function MerchantResourceInformationFromJSON(json) {
    return MerchantResourceInformationFromJSONTyped(json, false);
}
function MerchantResourceInformationFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'resourceType': json['resourceType'],
        'value': json['value'],
    };
}
function MerchantResourceInformationToJSON(json) {
    return MerchantResourceInformationToJSONTyped(json, false);
}
function MerchantResourceInformationToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'resourceType': value['resourceType'],
        'value': value['value'],
    };
}
const propertyValidationAttributesMap = {};
function validateMerchantResourceInformation(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
