// dana-node/dist/merchant_management/v1/models/MobileNoInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MobileNoInfoVerifiedEnum = void 0;
exports.instanceOfMobileNoInfo = instanceOfMobileNoInfo;
exports.MobileNoInfoFromJSON = MobileNoInfoFromJSON;
exports.MobileNoInfoFromJSONTyped = MobileNoInfoFromJSONTyped;
exports.MobileNoInfoToJSON = MobileNoInfoToJSON;
exports.MobileNoInfoToJSONTyped = MobileNoInfoToJSONTyped;
exports.validateMobileNoInfo = validateMobileNoInfo;
/**
 * @export
 */
exports.MobileNoInfoVerifiedEnum = {
    True: 'true',
    False: 'false'
};
/**
 * Check if a given object implements the MobileNoInfo interface.
 */
function instanceOfMobileNoInfo(value) {
    return true;
}
function MobileNoInfoFromJSON(json) {
    return MobileNoInfoFromJSONTyped(json, false);
}
function MobileNoInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'mobileNo': json['mobileNo'] == null ? undefined : json['mobileNo'],
        'mobileId': json['mobileId'] == null ? undefined : json['mobileId'],
        'verified': json['verified'] == null ? undefined : json['verified'],
    };
}
function MobileNoInfoToJSON(json) {
    return MobileNoInfoToJSONTyped(json, false);
}
function MobileNoInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'mobileNo': value['mobileNo'],
        'mobileId': value['mobileId'],
        'verified': value['verified'],
    };
}
const propertyValidationAttributesMap = {};
function validateMobileNoInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
