// dana-node/dist/merchant_management/v1/models/PicInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPicInfo = instanceOfPicInfo;
exports.PicInfoFromJSON = PicInfoFromJSON;
exports.PicInfoFromJSONTyped = PicInfoFromJSONTyped;
exports.PicInfoToJSON = PicInfoToJSON;
exports.PicInfoToJSONTyped = PicInfoToJSONTyped;
exports.validatePicInfo = validatePicInfo;
/**
 * Check if a given object implements the PicInfo interface.
 */
function instanceOfPicInfo(value) {
    return true;
}
function PicInfoFromJSON(json) {
    return PicInfoFromJSONTyped(json, false);
}
function PicInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'picName': json['picName'] == null ? undefined : json['picName'],
        'picPosition': json['picPosition'] == null ? undefined : json['picPosition'],
    };
}
function PicInfoToJSON(json) {
    return PicInfoToJSONTyped(json, false);
}
function PicInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'picName': value['picName'],
        'picPosition': value['picPosition'],
    };
}
const propertyValidationAttributesMap = {};
function validatePicInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
