// dana-node/dist/merchant_management/v1/models/QueryMerchantResourceResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryMerchantResourceResponseResponse = instanceOfQueryMerchantResourceResponseResponse;
exports.QueryMerchantResourceResponseResponseFromJSON = QueryMerchantResourceResponseResponseFromJSON;
exports.QueryMerchantResourceResponseResponseFromJSONTyped = QueryMerchantResourceResponseResponseFromJSONTyped;
exports.QueryMerchantResourceResponseResponseToJSON = QueryMerchantResourceResponseResponseToJSON;
exports.QueryMerchantResourceResponseResponseToJSONTyped = QueryMerchantResourceResponseResponseToJSONTyped;
exports.validateQueryMerchantResourceResponseResponse = validateQueryMerchantResourceResponseResponse;
const QueryMerchantResourceResponseResponseBody_1 = require("./QueryMerchantResourceResponseResponseBody");
const QueryMerchantResourceResponseResponseHead_1 = require("./QueryMerchantResourceResponseResponseHead");
/**
 * Check if a given object implements the QueryMerchantResourceResponseResponse interface.
 */
function instanceOfQueryMerchantResourceResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function QueryMerchantResourceResponseResponseFromJSON(json) {
    return QueryMerchantResourceResponseResponseFromJSONTyped(json, false);
}
function QueryMerchantResourceResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, QueryMerchantResourceResponseResponseHead_1.QueryMerchantResourceResponseResponseHeadFromJSON)(json['head']),
        'body': (0, QueryMerchantResourceResponseResponseBody_1.QueryMerchantResourceResponseResponseBodyFromJSON)(json['body']),
    };
}
function QueryMerchantResourceResponseResponseToJSON(json) {
    return QueryMerchantResourceResponseResponseToJSONTyped(json, false);
}
function QueryMerchantResourceResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, QueryMerchantResourceResponseResponseHead_1.QueryMerchantResourceResponseResponseHeadToJSON)(value['head']),
        'body': (0, QueryMerchantResourceResponseResponseBody_1.QueryMerchantResourceResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateQueryMerchantResourceResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, QueryMerchantResourceResponseResponseHead_1.validateQueryMerchantResourceResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, QueryMerchantResourceResponseResponseBody_1.validateQueryMerchantResourceResponseResponseBody)(value.body));
    return validationErrorContexts;
}
