// dana-node/dist/merchant_management/v1/models/QueryShopResponseResponseBody.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryShopResponseResponseBody = instanceOfQueryShopResponseResponseBody;
exports.QueryShopResponseResponseBodyFromJSON = QueryShopResponseResponseBodyFromJSON;
exports.QueryShopResponseResponseBodyFromJSONTyped = QueryShopResponseResponseBodyFromJSONTyped;
exports.QueryShopResponseResponseBodyToJSON = QueryShopResponseResponseBodyToJSON;
exports.QueryShopResponseResponseBodyToJSONTyped = QueryShopResponseResponseBodyToJSONTyped;
exports.validateQueryShopResponseResponseBody = validateQueryShopResponseResponseBody;
const ShopResourceInfo_1 = require("./ShopResourceInfo");
const ResultInfo_1 = require("./ResultInfo");
/**
 * Check if a given object implements the QueryShopResponseResponseBody interface.
 */
function instanceOfQueryShopResponseResponseBody(value) {
    if (!('resultInfo' in value) || value['resultInfo'] === undefined)
        return false;
    return true;
}
function QueryShopResponseResponseBodyFromJSON(json) {
    return QueryShopResponseResponseBodyFromJSONTyped(json, false);
}
function QueryShopResponseResponseBodyFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'resultInfo': (0, ResultInfo_1.ResultInfoFromJSON)(json['resultInfo']),
        'shopResourceInfo': json['shopResourceInfo'] == null ? undefined : (0, ShopResourceInfo_1.ShopResourceInfoFromJSON)(json['shopResourceInfo']),
    };
}
function QueryShopResponseResponseBodyToJSON(json) {
    return QueryShopResponseResponseBodyToJSONTyped(json, false);
}
function QueryShopResponseResponseBodyToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'resultInfo': (0, ResultInfo_1.ResultInfoToJSON)(value['resultInfo']),
        'shopResourceInfo': (0, ShopResourceInfo_1.ShopResourceInfoToJSON)(value['shopResourceInfo']),
    };
}
const propertyValidationAttributesMap = {};
function validateQueryShopResponseResponseBody(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, ResultInfo_1.validateResultInfo)(value.resultInfo));
    validationErrorContexts.push(...(0, ShopResourceInfo_1.validateShopResourceInfo)(value.shopResourceInfo));
    return validationErrorContexts;
}
