// dana-node/dist/merchant_management/v1/models/QueryDivisionResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryDivisionResponseResponse = instanceOfQueryDivisionResponseResponse;
exports.QueryDivisionResponseResponseFromJSON = QueryDivisionResponseResponseFromJSON;
exports.QueryDivisionResponseResponseFromJSONTyped = QueryDivisionResponseResponseFromJSONTyped;
exports.QueryDivisionResponseResponseToJSON = QueryDivisionResponseResponseToJSON;
exports.QueryDivisionResponseResponseToJSONTyped = QueryDivisionResponseResponseToJSONTyped;
exports.validateQueryDivisionResponseResponse = validateQueryDivisionResponseResponse;
const QueryDivisionResponseResponseBody_1 = require("./QueryDivisionResponseResponseBody");
const QueryDivisionResponseResponseHead_1 = require("./QueryDivisionResponseResponseHead");
/**
 * Check if a given object implements the QueryDivisionResponseResponse interface.
 */
function instanceOfQueryDivisionResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function QueryDivisionResponseResponseFromJSON(json) {
    return QueryDivisionResponseResponseFromJSONTyped(json, false);
}
function QueryDivisionResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, QueryDivisionResponseResponseHead_1.QueryDivisionResponseResponseHeadFromJSON)(json['head']),
        'body': (0, QueryDivisionResponseResponseBody_1.QueryDivisionResponseResponseBodyFromJSON)(json['body']),
    };
}
function QueryDivisionResponseResponseToJSON(json) {
    return QueryDivisionResponseResponseToJSONTyped(json, false);
}
function QueryDivisionResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, QueryDivisionResponseResponseHead_1.QueryDivisionResponseResponseHeadToJSON)(value['head']),
        'body': (0, QueryDivisionResponseResponseBody_1.QueryDivisionResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateQueryDivisionResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, QueryDivisionResponseResponseHead_1.validateQueryDivisionResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, QueryDivisionResponseResponseBody_1.validateQueryDivisionResponseResponseBody)(value.body));
    return validationErrorContexts;
}
