// dana-node/dist/merchant_management/v1/models/QueryShopResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryShopResponseResponse = instanceOfQueryShopResponseResponse;
exports.QueryShopResponseResponseFromJSON = QueryShopResponseResponseFromJSON;
exports.QueryShopResponseResponseFromJSONTyped = QueryShopResponseResponseFromJSONTyped;
exports.QueryShopResponseResponseToJSON = QueryShopResponseResponseToJSON;
exports.QueryShopResponseResponseToJSONTyped = QueryShopResponseResponseToJSONTyped;
exports.validateQueryShopResponseResponse = validateQueryShopResponseResponse;
const QueryShopResponseResponseBody_1 = require("./QueryShopResponseResponseBody");
const QueryShopResponseResponseHead_1 = require("./QueryShopResponseResponseHead");
/**
 * Check if a given object implements the QueryShopResponseResponse interface.
 */
function instanceOfQueryShopResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function QueryShopResponseResponseFromJSON(json) {
    return QueryShopResponseResponseFromJSONTyped(json, false);
}
function QueryShopResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, QueryShopResponseResponseHead_1.QueryShopResponseResponseHeadFromJSON)(json['head']),
        'body': (0, QueryShopResponseResponseBody_1.QueryShopResponseResponseBodyFromJSON)(json['body']),
    };
}
function QueryShopResponseResponseToJSON(json) {
    return QueryShopResponseResponseToJSONTyped(json, false);
}
function QueryShopResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, QueryShopResponseResponseHead_1.QueryShopResponseResponseHeadToJSON)(value['head']),
        'body': (0, QueryShopResponseResponseBody_1.QueryShopResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateQueryShopResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, QueryShopResponseResponseHead_1.validateQueryShopResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, QueryShopResponseResponseBody_1.validateQueryShopResponseResponseBody)(value.body));
    return validationErrorContexts;
}
