// dana-node/dist/merchant_management/v1/models/QueryMerchantResourceResponseResponseHead.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryMerchantResourceResponseResponseHeadFunctionEnum = void 0;
exports.instanceOfQueryMerchantResourceResponseResponseHead = instanceOfQueryMerchantResourceResponseResponseHead;
exports.QueryMerchantResourceResponseResponseHeadFromJSON = QueryMerchantResourceResponseResponseHeadFromJSON;
exports.QueryMerchantResourceResponseResponseHeadFromJSONTyped = QueryMerchantResourceResponseResponseHeadFromJSONTyped;
exports.QueryMerchantResourceResponseResponseHeadToJSON = QueryMerchantResourceResponseResponseHeadToJSON;
exports.QueryMerchantResourceResponseResponseHeadToJSONTyped = QueryMerchantResourceResponseResponseHeadToJSONTyped;
exports.validateQueryMerchantResourceResponseResponseHead = validateQueryMerchantResourceResponseResponseHead;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.QueryMerchantResourceResponseResponseHeadFunctionEnum = {
    DanaMerchantQueryMerchantResource: 'dana.merchant.queryMerchantResource'
};
/**
 * Check if a given object implements the QueryMerchantResourceResponseResponseHead interface.
 */
function instanceOfQueryMerchantResourceResponseResponseHead(value) {
    return true;
}
function QueryMerchantResourceResponseResponseHeadFromJSON(json) {
    return QueryMerchantResourceResponseResponseHeadFromJSONTyped(json, false);
}
function QueryMerchantResourceResponseResponseHeadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'version': json['version'] == null ? undefined : json['version'],
        '_function': json['function'] == null ? undefined : json['function'],
        'clientId': json['clientId'] == null ? undefined : json['clientId'],
        'respTime': json['respTime'] == null ? undefined : json['respTime'],
        'reqMsgId': json['reqMsgId'] == null ? undefined : json['reqMsgId'],
    };
}
function QueryMerchantResourceResponseResponseHeadToJSON(json) {
    return QueryMerchantResourceResponseResponseHeadToJSONTyped(json, false);
}
function QueryMerchantResourceResponseResponseHeadToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'version': value['version'],
        'function': value['_function'],
        'clientId': value['clientId'],
        'respTime': value['respTime'],
        'reqMsgId': value['reqMsgId'],
    };
}
const propertyValidationAttributesMap = {
    version: {
        maxLength: 8,
    },
    _function: {
        maxLength: 128,
    },
    clientId: {
        maxLength: 32,
    },
    reqMsgId: {
        maxLength: 64,
    },
};
function validateQueryMerchantResourceResponseResponseHead(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('version', value.version, propertyValidationAttributesMap['version']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('_function', value._function, propertyValidationAttributesMap['_function']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientId', value.clientId, propertyValidationAttributesMap['clientId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reqMsgId', value.reqMsgId, propertyValidationAttributesMap['reqMsgId']));
    return validationErrorContexts;
}
