// dana-node/dist/merchant_management/v1/models/QueryDivisionResponseResponseHead.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryDivisionResponseResponseHeadFunctionEnum = void 0;
exports.instanceOfQueryDivisionResponseResponseHead = instanceOfQueryDivisionResponseResponseHead;
exports.QueryDivisionResponseResponseHeadFromJSON = QueryDivisionResponseResponseHeadFromJSON;
exports.QueryDivisionResponseResponseHeadFromJSONTyped = QueryDivisionResponseResponseHeadFromJSONTyped;
exports.QueryDivisionResponseResponseHeadToJSON = QueryDivisionResponseResponseHeadToJSON;
exports.QueryDivisionResponseResponseHeadToJSONTyped = QueryDivisionResponseResponseHeadToJSONTyped;
exports.validateQueryDivisionResponseResponseHead = validateQueryDivisionResponseResponseHead;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.QueryDivisionResponseResponseHeadFunctionEnum = {
    DanaMerchantDivisionQueryDivision: 'dana.merchant.division.queryDivision'
};
/**
 * Check if a given object implements the QueryDivisionResponseResponseHead interface.
 */
function instanceOfQueryDivisionResponseResponseHead(value) {
    return true;
}
function QueryDivisionResponseResponseHeadFromJSON(json) {
    return QueryDivisionResponseResponseHeadFromJSONTyped(json, false);
}
function QueryDivisionResponseResponseHeadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'version': json['version'] == null ? undefined : json['version'],
        '_function': json['function'] == null ? undefined : json['function'],
        'clientId': json['clientId'] == null ? undefined : json['clientId'],
        'clientSecret': json['clientSecret'] == null ? undefined : json['clientSecret'],
        'respTime': json['respTime'] == null ? undefined : json['respTime'],
        'reqMsgId': json['reqMsgId'] == null ? undefined : json['reqMsgId'],
        'reserve': json['reserve'] == null ? undefined : json['reserve'],
    };
}
function QueryDivisionResponseResponseHeadToJSON(json) {
    return QueryDivisionResponseResponseHeadToJSONTyped(json, false);
}
function QueryDivisionResponseResponseHeadToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'version': value['version'],
        'function': value['_function'],
        'clientId': value['clientId'],
        'clientSecret': value['clientSecret'],
        'respTime': value['respTime'],
        'reqMsgId': value['reqMsgId'],
        'reserve': value['reserve'],
    };
}
const propertyValidationAttributesMap = {
    version: {
        maxLength: 8,
    },
    _function: {
        maxLength: 128,
    },
    clientId: {
        maxLength: 36,
    },
    clientSecret: {
        maxLength: 64,
    },
    respTime: {
        maxLength: 25,
    },
    reqMsgId: {
        maxLength: 64,
    },
    reserve: {
        maxLength: 256,
    },
};
function validateQueryDivisionResponseResponseHead(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('version', value.version, propertyValidationAttributesMap['version']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('_function', value._function, propertyValidationAttributesMap['_function']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientId', value.clientId, propertyValidationAttributesMap['clientId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientSecret', value.clientSecret, propertyValidationAttributesMap['clientSecret']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('respTime', value.respTime, propertyValidationAttributesMap['respTime']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reqMsgId', value.reqMsgId, propertyValidationAttributesMap['reqMsgId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reserve', value.reserve, propertyValidationAttributesMap['reserve']));
    return validationErrorContexts;
}
