// dana-node/dist/merchant_management/v1/models/QueryShopResponseResponseHead.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryShopResponseResponseHeadFunctionEnum = void 0;
exports.instanceOfQueryShopResponseResponseHead = instanceOfQueryShopResponseResponseHead;
exports.QueryShopResponseResponseHeadFromJSON = QueryShopResponseResponseHeadFromJSON;
exports.QueryShopResponseResponseHeadFromJSONTyped = QueryShopResponseResponseHeadFromJSONTyped;
exports.QueryShopResponseResponseHeadToJSON = QueryShopResponseResponseHeadToJSON;
exports.QueryShopResponseResponseHeadToJSONTyped = QueryShopResponseResponseHeadToJSONTyped;
exports.validateQueryShopResponseResponseHead = validateQueryShopResponseResponseHead;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.QueryShopResponseResponseHeadFunctionEnum = {
    DanaMerchantShopQueryShop: 'dana.merchant.shop.queryShop'
};
/**
 * Check if a given object implements the QueryShopResponseResponseHead interface.
 */
function instanceOfQueryShopResponseResponseHead(value) {
    return true;
}
function QueryShopResponseResponseHeadFromJSON(json) {
    return QueryShopResponseResponseHeadFromJSONTyped(json, false);
}
function QueryShopResponseResponseHeadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'version': json['version'] == null ? undefined : json['version'],
        '_function': json['function'] == null ? undefined : json['function'],
        'clientId': json['clientId'] == null ? undefined : json['clientId'],
        'respTime': json['respTime'] == null ? undefined : json['respTime'],
        'reqMsgId': json['reqMsgId'] == null ? undefined : json['reqMsgId'],
        'clientSecret': json['clientSecret'] == null ? undefined : json['clientSecret'],
        'reserve': json['reserve'] == null ? undefined : json['reserve'],
    };
}
function QueryShopResponseResponseHeadToJSON(json) {
    return QueryShopResponseResponseHeadToJSONTyped(json, false);
}
function QueryShopResponseResponseHeadToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'version': value['version'],
        'function': value['_function'],
        'clientId': value['clientId'],
        'respTime': value['respTime'],
        'reqMsgId': value['reqMsgId'],
        'clientSecret': value['clientSecret'],
        'reserve': value['reserve'],
    };
}
const propertyValidationAttributesMap = {
    version: {
        maxLength: 8,
    },
    _function: {
        maxLength: 128,
    },
    clientId: {
        maxLength: 36,
    },
    respTime: {
        maxLength: 25,
    },
    reqMsgId: {
        maxLength: 64,
    },
    clientSecret: {
        maxLength: 64,
    },
    reserve: {
        maxLength: 256,
    },
};
function validateQueryShopResponseResponseHead(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('version', value.version, propertyValidationAttributesMap['version']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('_function', value._function, propertyValidationAttributesMap['_function']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientId', value.clientId, propertyValidationAttributesMap['clientId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('respTime', value.respTime, propertyValidationAttributesMap['respTime']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reqMsgId', value.reqMsgId, propertyValidationAttributesMap['reqMsgId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientSecret', value.clientSecret, propertyValidationAttributesMap['clientSecret']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reserve', value.reserve, propertyValidationAttributesMap['reserve']));
    return validationErrorContexts;
}
