// dana-node/dist/merchant_management/v1/models/QueryDivisionResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryDivisionResponse = instanceOfQueryDivisionResponse;
exports.QueryDivisionResponseFromJSON = QueryDivisionResponseFromJSON;
exports.QueryDivisionResponseFromJSONTyped = QueryDivisionResponseFromJSONTyped;
exports.QueryDivisionResponseToJSON = QueryDivisionResponseToJSON;
exports.QueryDivisionResponseToJSONTyped = QueryDivisionResponseToJSONTyped;
exports.validateQueryDivisionResponse = validateQueryDivisionResponse;
const QueryDivisionResponseResponse_1 = require("./QueryDivisionResponseResponse");
/**
 * Check if a given object implements the QueryDivisionResponse interface.
 */
function instanceOfQueryDivisionResponse(value) {
    if (!('response' in value) || value['response'] === undefined)
        return false;
    return true;
}
function QueryDivisionResponseFromJSON(json) {
    return QueryDivisionResponseFromJSONTyped(json, false);
}
function QueryDivisionResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'response': (0, QueryDivisionResponseResponse_1.QueryDivisionResponseResponseFromJSON)(json['response']),
        'signature': json['signature'] == null ? undefined : json['signature'],
    };
}
function QueryDivisionResponseToJSON(json) {
    return QueryDivisionResponseToJSONTyped(json, false);
}
function QueryDivisionResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'response': (0, QueryDivisionResponseResponse_1.QueryDivisionResponseResponseToJSON)(value['response']),
        'signature': value['signature'],
    };
}
const propertyValidationAttributesMap = {};
function validateQueryDivisionResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, QueryDivisionResponseResponse_1.validateQueryDivisionResponseResponse)(value.response));
    return validationErrorContexts;
}
