// dana-node/dist/merchant_management/v1/models/QueryMerchantResourceRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryMerchantResourceRequestMerchantResourceInfoListEnum = void 0;
exports.instanceOfQueryMerchantResourceRequest = instanceOfQueryMerchantResourceRequest;
exports.QueryMerchantResourceRequestFromJSON = QueryMerchantResourceRequestFromJSON;
exports.QueryMerchantResourceRequestFromJSONTyped = QueryMerchantResourceRequestFromJSONTyped;
exports.QueryMerchantResourceRequestToJSON = QueryMerchantResourceRequestToJSON;
exports.QueryMerchantResourceRequestToJSONTyped = QueryMerchantResourceRequestToJSONTyped;
exports.validateQueryMerchantResourceRequest = validateQueryMerchantResourceRequest;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.QueryMerchantResourceRequestMerchantResourceInfoListEnum = {
    MerchantDepositBalance: 'MERCHANT_DEPOSIT_BALANCE',
    MerchantAvailableBalance: 'MERCHANT_AVAILABLE_BALANCE',
    MerchantTotalBalance: 'MERCHANT_TOTAL_BALANCE'
};
/**
 * Check if a given object implements the QueryMerchantResourceRequest interface.
 */
function instanceOfQueryMerchantResourceRequest(value) {
    if (!('requestMerchantId' in value) || value['requestMerchantId'] === undefined)
        return false;
    if (!('merchantResourceInfoList' in value) || value['merchantResourceInfoList'] === undefined)
        return false;
    return true;
}
function QueryMerchantResourceRequestFromJSON(json) {
    return QueryMerchantResourceRequestFromJSONTyped(json, false);
}
function QueryMerchantResourceRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'requestMerchantId': json['requestMerchantId'],
        'merchantResourceInfoList': json['merchantResourceInfoList'],
    };
}
function QueryMerchantResourceRequestToJSON(json) {
    return QueryMerchantResourceRequestToJSONTyped(json, false);
}
function QueryMerchantResourceRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'requestMerchantId': value['requestMerchantId'],
        'merchantResourceInfoList': value['merchantResourceInfoList'],
    };
}
const propertyValidationAttributesMap = {
    merchantResourceInfoList: {
        minItems: 1,
        uniqueItems: false,
    },
};
function validateQueryMerchantResourceRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantResourceInfoList', value.merchantResourceInfoList, propertyValidationAttributesMap['merchantResourceInfoList']));
    return validationErrorContexts;
}
