// dana-node/dist/merchant_management/v1/models/UpdateShopResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateShopResponseResponse = instanceOfUpdateShopResponseResponse;
exports.UpdateShopResponseResponseFromJSON = UpdateShopResponseResponseFromJSON;
exports.UpdateShopResponseResponseFromJSONTyped = UpdateShopResponseResponseFromJSONTyped;
exports.UpdateShopResponseResponseToJSON = UpdateShopResponseResponseToJSON;
exports.UpdateShopResponseResponseToJSONTyped = UpdateShopResponseResponseToJSONTyped;
exports.validateUpdateShopResponseResponse = validateUpdateShopResponseResponse;
const UpdateDivisionResponseResponseBody_1 = require("./UpdateDivisionResponseResponseBody");
const UpdateShopResponseResponseHead_1 = require("./UpdateShopResponseResponseHead");
/**
 * Check if a given object implements the UpdateShopResponseResponse interface.
 */
function instanceOfUpdateShopResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function UpdateShopResponseResponseFromJSON(json) {
    return UpdateShopResponseResponseFromJSONTyped(json, false);
}
function UpdateShopResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, UpdateShopResponseResponseHead_1.UpdateShopResponseResponseHeadFromJSON)(json['head']),
        'body': (0, UpdateDivisionResponseResponseBody_1.UpdateDivisionResponseResponseBodyFromJSON)(json['body']),
    };
}
function UpdateShopResponseResponseToJSON(json) {
    return UpdateShopResponseResponseToJSONTyped(json, false);
}
function UpdateShopResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, UpdateShopResponseResponseHead_1.UpdateShopResponseResponseHeadToJSON)(value['head']),
        'body': (0, UpdateDivisionResponseResponseBody_1.UpdateDivisionResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateUpdateShopResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, UpdateShopResponseResponseHead_1.validateUpdateShopResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, UpdateDivisionResponseResponseBody_1.validateUpdateDivisionResponseResponseBody)(value.body));
    return validationErrorContexts;
}
