// dana-node/dist/merchant_management/v1/models/UpdateShopRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateShopRequestShopBizTypeEnum = exports.UpdateShopRequestShopOwningEnum = exports.UpdateShopRequestBusinessEntityEnum = exports.UpdateShopRequestOwnerIdTypeEnum = exports.UpdateShopRequestLoyaltyEnum = exports.UpdateShopRequestSizeTypeEnum = exports.UpdateShopRequestShopIdTypeEnum = void 0;
exports.instanceOfUpdateShopRequest = instanceOfUpdateShopRequest;
exports.UpdateShopRequestFromJSON = UpdateShopRequestFromJSON;
exports.UpdateShopRequestFromJSONTyped = UpdateShopRequestFromJSONTyped;
exports.UpdateShopRequestToJSON = UpdateShopRequestToJSON;
exports.UpdateShopRequestToJSONTyped = UpdateShopRequestToJSONTyped;
exports.validateUpdateShopRequest = validateUpdateShopRequest;
const runtime_1 = require("../../../runtime");
const BusinessDocs_1 = require("./BusinessDocs");
const UserName_1 = require("./UserName");
const MobileNoInfo_1 = require("./MobileNoInfo");
const PicInfo_1 = require("./PicInfo");
const AddressInfo_1 = require("./AddressInfo");
/**
 * @export
 */
exports.UpdateShopRequestShopIdTypeEnum = {
    InnerId: 'INNER_ID',
    ExternalId: 'EXTERNAL_ID'
};
/**
 * @export
 */
exports.UpdateShopRequestSizeTypeEnum = {
    Umi: 'UMI',
    Uke: 'UKE',
    Ume: 'UME',
    Ube: 'UBE'
};
/**
 * @export
 */
exports.UpdateShopRequestLoyaltyEnum = {
    True: 'true',
    False: 'false'
};
/**
 * @export
 */
exports.UpdateShopRequestOwnerIdTypeEnum = {
    Ktp: 'KTP',
    Sim: 'SIM',
    Passport: 'PASSPORT',
    Siup: 'SIUP',
    Nib: 'NIB'
};
/**
 * @export
 */
exports.UpdateShopRequestBusinessEntityEnum = {
    Pt: 'pt',
    Cv: 'cv',
    Individu: 'individu',
    UsahaDagang: 'usaha_dagang',
    Yayasan: 'yayasan',
    Koperasi: 'koperasi'
};
/**
 * @export
 */
exports.UpdateShopRequestShopOwningEnum = {
    DirectOwned: 'DIRECT_OWNED',
    Franchised: 'FRANCHISED'
};
/**
 * @export
 */
exports.UpdateShopRequestShopBizTypeEnum = {
    Online: 'ONLINE',
    Offline: 'OFFLINE'
};
/**
 * Check if a given object implements the UpdateShopRequest interface.
 */
function instanceOfUpdateShopRequest(value) {
    if (!('shopId' in value) || value['shopId'] === undefined)
        return false;
    if (!('merchantId' in value) || value['merchantId'] === undefined)
        return false;
    if (!('shopIdType' in value) || value['shopIdType'] === undefined)
        return false;
    if (!('shopAddress' in value) || value['shopAddress'] === undefined)
        return false;
    return true;
}
function UpdateShopRequestFromJSON(json) {
    return UpdateShopRequestFromJSONTyped(json, false);
}
function UpdateShopRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'shopId': json['shopId'],
        'merchantId': json['merchantId'],
        'shopIdType': json['shopIdType'],
        'mainName': json['mainName'] == null ? undefined : json['mainName'],
        'shopAddress': (0, AddressInfo_1.AddressInfoFromJSON)(json['shopAddress']),
        'shopDesc': json['shopDesc'] == null ? undefined : json['shopDesc'],
        'newExternalShopId': json['newExternalShopId'] == null ? undefined : json['newExternalShopId'],
        'mccCodes': json['mccCodes'] == null ? undefined : json['mccCodes'],
        'logoUrlMap': json['logoUrlMap'] == null ? undefined : json['logoUrlMap'],
        'extInfo': json['extInfo'] == null ? undefined : json['extInfo'],
        'sizeType': json['sizeType'] == null ? undefined : json['sizeType'],
        'ln': json['ln'] == null ? undefined : json['ln'],
        'lat': json['lat'] == null ? undefined : json['lat'],
        'loyalty': json['loyalty'] == null ? undefined : json['loyalty'],
        'ownerAddress': json['ownerAddress'] == null ? undefined : (0, AddressInfo_1.AddressInfoFromJSON)(json['ownerAddress']),
        'ownerName': json['ownerName'] == null ? undefined : (0, UserName_1.UserNameFromJSON)(json['ownerName']),
        'ownerPhoneNumber': json['ownerPhoneNumber'] == null ? undefined : (0, MobileNoInfo_1.MobileNoInfoFromJSON)(json['ownerPhoneNumber']),
        'ownerIdType': json['ownerIdType'] == null ? undefined : json['ownerIdType'],
        'ownerIdNo': json['ownerIdNo'] == null ? undefined : json['ownerIdNo'],
        'deviceNumber': json['deviceNumber'] == null ? undefined : json['deviceNumber'],
        'posNumber': json['posNumber'] == null ? undefined : json['posNumber'],
        'apiVersion': json['apiVersion'] == null ? undefined : json['apiVersion'],
        'businessEntity': json['businessEntity'] == null ? undefined : json['businessEntity'],
        'shopOwning': json['shopOwning'] == null ? undefined : json['shopOwning'],
        'shopBizType': json['shopBizType'] == null ? undefined : json['shopBizType'],
        'businessDocs': json['businessDocs'] == null ? undefined : (json['businessDocs'].map(BusinessDocs_1.BusinessDocsFromJSON)),
        'businessEndDate': json['businessEndDate'] == null ? undefined : json['businessEndDate'],
        'taxNo': json['taxNo'] == null ? undefined : json['taxNo'],
        'taxAddress': json['taxAddress'] == null ? undefined : (0, AddressInfo_1.AddressInfoFromJSON)(json['taxAddress']),
        'brandName': json['brandName'] == null ? undefined : json['brandName'],
        'directorPics': json['directorPics'] == null ? undefined : (json['directorPics'].map(PicInfo_1.PicInfoFromJSON)),
        'nonDirectorPics': json['nonDirectorPics'] == null ? undefined : (json['nonDirectorPics'].map(PicInfo_1.PicInfoFromJSON)),
    };
}
function UpdateShopRequestToJSON(json) {
    return UpdateShopRequestToJSONTyped(json, false);
}
function UpdateShopRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'shopId': value['shopId'],
        'merchantId': value['merchantId'],
        'shopIdType': value['shopIdType'],
        'mainName': value['mainName'],
        'shopAddress': (0, AddressInfo_1.AddressInfoToJSON)(value['shopAddress']),
        'shopDesc': value['shopDesc'],
        'newExternalShopId': value['newExternalShopId'],
        'mccCodes': value['mccCodes'],
        'logoUrlMap': value['logoUrlMap'],
        'extInfo': value['extInfo'],
        'sizeType': value['sizeType'],
        'ln': value['ln'],
        'lat': value['lat'],
        'loyalty': value['loyalty'],
        'ownerAddress': (0, AddressInfo_1.AddressInfoToJSON)(value['ownerAddress']),
        'ownerName': (0, UserName_1.UserNameToJSON)(value['ownerName']),
        'ownerPhoneNumber': (0, MobileNoInfo_1.MobileNoInfoToJSON)(value['ownerPhoneNumber']),
        'ownerIdType': value['ownerIdType'],
        'ownerIdNo': value['ownerIdNo'],
        'deviceNumber': value['deviceNumber'],
        'posNumber': value['posNumber'],
        'apiVersion': value['apiVersion'],
        'businessEntity': value['businessEntity'],
        'shopOwning': value['shopOwning'],
        'shopBizType': value['shopBizType'],
        'businessDocs': value['businessDocs'] == null ? undefined : (value['businessDocs'].map(BusinessDocs_1.BusinessDocsToJSON)),
        'businessEndDate': value['businessEndDate'],
        'taxNo': value['taxNo'],
        'taxAddress': (0, AddressInfo_1.AddressInfoToJSON)(value['taxAddress']),
        'brandName': value['brandName'],
        'directorPics': value['directorPics'] == null ? undefined : (value['directorPics'].map(PicInfo_1.PicInfoToJSON)),
        'nonDirectorPics': value['nonDirectorPics'] == null ? undefined : (value['nonDirectorPics'].map(PicInfo_1.PicInfoToJSON)),
    };
}
const propertyValidationAttributesMap = {
    merchantId: {
        maxLength: 21,
    },
    mainName: {
        maxLength: 256,
    },
    shopDesc: {
        maxLength: 1024,
    },
    newExternalShopId: {
        maxLength: 64,
    },
    extInfo: {},
    ln: {
        maxLength: 10,
    },
    lat: {
        maxLength: 10,
    },
    apiVersion: {
        maxLength: 8,
    },
    businessEndDate: {
        maxLength: 10,
    },
    taxNo: {
        pattern: new RegExp('/^[0-9]{15}$/'.slice(1, -1)),
    },
    brandName: {
        maxLength: 256,
    },
};
function validateUpdateShopRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantId', value.merchantId, propertyValidationAttributesMap['merchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('mainName', value.mainName, propertyValidationAttributesMap['mainName']));
    validationErrorContexts.push(...(0, AddressInfo_1.validateAddressInfo)(value.shopAddress));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('shopDesc', value.shopDesc, propertyValidationAttributesMap['shopDesc']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('newExternalShopId', value.newExternalShopId, propertyValidationAttributesMap['newExternalShopId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extInfo', value.extInfo, propertyValidationAttributesMap['extInfo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('ln', value.ln, propertyValidationAttributesMap['ln']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('lat', value.lat, propertyValidationAttributesMap['lat']));
    validationErrorContexts.push(...(0, AddressInfo_1.validateAddressInfo)(value.ownerAddress));
    validationErrorContexts.push(...(0, UserName_1.validateUserName)(value.ownerName));
    validationErrorContexts.push(...(0, MobileNoInfo_1.validateMobileNoInfo)(value.ownerPhoneNumber));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('apiVersion', value.apiVersion, propertyValidationAttributesMap['apiVersion']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('businessEndDate', value.businessEndDate, propertyValidationAttributesMap['businessEndDate']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('taxNo', value.taxNo, propertyValidationAttributesMap['taxNo']));
    validationErrorContexts.push(...(0, AddressInfo_1.validateAddressInfo)(value.taxAddress));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('brandName', value.brandName, propertyValidationAttributesMap['brandName']));
    return validationErrorContexts;
}
