// dana-node/dist/merchant_management/v1/models/UpdateDivisionResponseResponseHead.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDivisionResponseResponseHeadFunctionEnum = void 0;
exports.instanceOfUpdateDivisionResponseResponseHead = instanceOfUpdateDivisionResponseResponseHead;
exports.UpdateDivisionResponseResponseHeadFromJSON = UpdateDivisionResponseResponseHeadFromJSON;
exports.UpdateDivisionResponseResponseHeadFromJSONTyped = UpdateDivisionResponseResponseHeadFromJSONTyped;
exports.UpdateDivisionResponseResponseHeadToJSON = UpdateDivisionResponseResponseHeadToJSON;
exports.UpdateDivisionResponseResponseHeadToJSONTyped = UpdateDivisionResponseResponseHeadToJSONTyped;
exports.validateUpdateDivisionResponseResponseHead = validateUpdateDivisionResponseResponseHead;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.UpdateDivisionResponseResponseHeadFunctionEnum = {
    DanaMerchantDivisionUpdateDivision: 'dana.merchant.division.updateDivision'
};
/**
 * Check if a given object implements the UpdateDivisionResponseResponseHead interface.
 */
function instanceOfUpdateDivisionResponseResponseHead(value) {
    return true;
}
function UpdateDivisionResponseResponseHeadFromJSON(json) {
    return UpdateDivisionResponseResponseHeadFromJSONTyped(json, false);
}
function UpdateDivisionResponseResponseHeadFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'version': json['version'] == null ? undefined : json['version'],
        '_function': json['function'] == null ? undefined : json['function'],
        'clientId': json['clientId'] == null ? undefined : json['clientId'],
        'clientSecret': json['clientSecret'] == null ? undefined : json['clientSecret'],
        'respTime': json['respTime'] == null ? undefined : json['respTime'],
        'reqMsgId': json['reqMsgId'] == null ? undefined : json['reqMsgId'],
        'reserve': json['reserve'] == null ? undefined : json['reserve'],
    };
}
function UpdateDivisionResponseResponseHeadToJSON(json) {
    return UpdateDivisionResponseResponseHeadToJSONTyped(json, false);
}
function UpdateDivisionResponseResponseHeadToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'version': value['version'],
        'function': value['_function'],
        'clientId': value['clientId'],
        'clientSecret': value['clientSecret'],
        'respTime': value['respTime'],
        'reqMsgId': value['reqMsgId'],
        'reserve': value['reserve'],
    };
}
const propertyValidationAttributesMap = {
    version: {
        maxLength: 8,
    },
    _function: {
        maxLength: 128,
    },
    clientId: {
        maxLength: 36,
    },
    clientSecret: {
        maxLength: 64,
    },
    respTime: {
        maxLength: 25,
    },
    reqMsgId: {
        maxLength: 64,
    },
    reserve: {
        maxLength: 256,
    },
};
function validateUpdateDivisionResponseResponseHead(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('version', value.version, propertyValidationAttributesMap['version']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('_function', value._function, propertyValidationAttributesMap['_function']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientId', value.clientId, propertyValidationAttributesMap['clientId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientSecret', value.clientSecret, propertyValidationAttributesMap['clientSecret']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('respTime', value.respTime, propertyValidationAttributesMap['respTime']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reqMsgId', value.reqMsgId, propertyValidationAttributesMap['reqMsgId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reserve', value.reserve, propertyValidationAttributesMap['reserve']));
    return validationErrorContexts;
}
