// dana-node/dist/merchant_management/v1/models/UpdateShopResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateShopResponse = instanceOfUpdateShopResponse;
exports.UpdateShopResponseFromJSON = UpdateShopResponseFromJSON;
exports.UpdateShopResponseFromJSONTyped = UpdateShopResponseFromJSONTyped;
exports.UpdateShopResponseToJSON = UpdateShopResponseToJSON;
exports.UpdateShopResponseToJSONTyped = UpdateShopResponseToJSONTyped;
exports.validateUpdateShopResponse = validateUpdateShopResponse;
const UpdateShopResponseResponse_1 = require("./UpdateShopResponseResponse");
/**
 * Check if a given object implements the UpdateShopResponse interface.
 */
function instanceOfUpdateShopResponse(value) {
    if (!('response' in value) || value['response'] === undefined)
        return false;
    return true;
}
function UpdateShopResponseFromJSON(json) {
    return UpdateShopResponseFromJSONTyped(json, false);
}
function UpdateShopResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'response': (0, UpdateShopResponseResponse_1.UpdateShopResponseResponseFromJSON)(json['response']),
        'signature': json['signature'] == null ? undefined : json['signature'],
    };
}
function UpdateShopResponseToJSON(json) {
    return UpdateShopResponseToJSONTyped(json, false);
}
function UpdateShopResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'response': (0, UpdateShopResponseResponse_1.UpdateShopResponseResponseToJSON)(value['response']),
        'signature': value['signature'],
    };
}
const propertyValidationAttributesMap = {};
function validateUpdateShopResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, UpdateShopResponseResponse_1.validateUpdateShopResponseResponse)(value.response));
    return validationErrorContexts;
}
