// dana-node/dist/merchant_management/v1/models/UpdateDivisionResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateDivisionResponseResponse = instanceOfUpdateDivisionResponseResponse;
exports.UpdateDivisionResponseResponseFromJSON = UpdateDivisionResponseResponseFromJSON;
exports.UpdateDivisionResponseResponseFromJSONTyped = UpdateDivisionResponseResponseFromJSONTyped;
exports.UpdateDivisionResponseResponseToJSON = UpdateDivisionResponseResponseToJSON;
exports.UpdateDivisionResponseResponseToJSONTyped = UpdateDivisionResponseResponseToJSONTyped;
exports.validateUpdateDivisionResponseResponse = validateUpdateDivisionResponseResponse;
const UpdateDivisionResponseResponseBody_1 = require("./UpdateDivisionResponseResponseBody");
const UpdateDivisionResponseResponseHead_1 = require("./UpdateDivisionResponseResponseHead");
/**
 * Check if a given object implements the UpdateDivisionResponseResponse interface.
 */
function instanceOfUpdateDivisionResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function UpdateDivisionResponseResponseFromJSON(json) {
    return UpdateDivisionResponseResponseFromJSONTyped(json, false);
}
function UpdateDivisionResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, UpdateDivisionResponseResponseHead_1.UpdateDivisionResponseResponseHeadFromJSON)(json['head']),
        'body': (0, UpdateDivisionResponseResponseBody_1.UpdateDivisionResponseResponseBodyFromJSON)(json['body']),
    };
}
function UpdateDivisionResponseResponseToJSON(json) {
    return UpdateDivisionResponseResponseToJSONTyped(json, false);
}
function UpdateDivisionResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, UpdateDivisionResponseResponseHead_1.UpdateDivisionResponseResponseHeadToJSON)(value['head']),
        'body': (0, UpdateDivisionResponseResponseBody_1.UpdateDivisionResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateUpdateDivisionResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, UpdateDivisionResponseResponseHead_1.validateUpdateDivisionResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, UpdateDivisionResponseResponseBody_1.validateUpdateDivisionResponseResponseBody)(value.body));
    return validationErrorContexts;
}
