// dana-node/dist/merchant_management/v1/models/UserName.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUserName = instanceOfUserName;
exports.UserNameFromJSON = UserNameFromJSON;
exports.UserNameFromJSONTyped = UserNameFromJSONTyped;
exports.UserNameToJSON = UserNameToJSON;
exports.UserNameToJSONTyped = UserNameToJSONTyped;
exports.validateUserName = validateUserName;
/**
 * Check if a given object implements the UserName interface.
 */
function instanceOfUserName(value) {
    return true;
}
function UserNameFromJSON(json) {
    return UserNameFromJSONTyped(json, false);
}
function UserNameFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'firstName': json['firstName'] == null ? undefined : json['firstName'],
        'lastName': json['lastName'] == null ? undefined : json['lastName'],
    };
}
function UserNameToJSON(json) {
    return UserNameToJSONTyped(json, false);
}
function UserNameToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'firstName': value['firstName'],
        'lastName': value['lastName'],
    };
}
const propertyValidationAttributesMap = {};
function validateUserName(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
