// dana-node/dist/webhook/v1/FinishNotifyRequestAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFinishNotifyRequestAdditionalInfo = instanceOfFinishNotifyRequestAdditionalInfo;
exports.FinishNotifyRequestAdditionalInfoFromJSON = FinishNotifyRequestAdditionalInfoFromJSON;
exports.FinishNotifyRequestAdditionalInfoFromJSONTyped = FinishNotifyRequestAdditionalInfoFromJSONTyped;
exports.FinishNotifyRequestAdditionalInfoToJSON = FinishNotifyRequestAdditionalInfoToJSON;
exports.FinishNotifyRequestAdditionalInfoToJSONTyped = FinishNotifyRequestAdditionalInfoToJSONTyped;
exports.validateFinishNotifyRequestAdditionalInfo = validateFinishNotifyRequestAdditionalInfo;
const runtime_1 = require("../../runtime");
const ShopInfo_1 = require("./ShopInfo");
const FinishNotifyPaymentInfo_1 = require("./FinishNotifyPaymentInfo");
/**
 * Check if a given object implements the FinishNotifyRequestAdditionalInfo interface.
 */
function instanceOfFinishNotifyRequestAdditionalInfo(value) {
    return true;
}
function FinishNotifyRequestAdditionalInfoFromJSON(json) {
    return FinishNotifyRequestAdditionalInfoFromJSONTyped(json, false);
}
function FinishNotifyRequestAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'paymentInfo': json['paymentInfo'] == null ? undefined : (0, FinishNotifyPaymentInfo_1.FinishNotifyPaymentInfoFromJSON)(json['paymentInfo']),
        'shopInfo': json['shopInfo'] == null ? undefined : (0, ShopInfo_1.ShopInfoFromJSON)(json['shopInfo']),
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'extendInfoClosedReason': json['extendInfo.closedReason'] == null ? undefined : json['extendInfo.closedReason'],
        'paidTime': json['paidTime'] == null ? undefined : json['paidTime'],
    };
}
function FinishNotifyRequestAdditionalInfoToJSON(json) {
    return FinishNotifyRequestAdditionalInfoToJSONTyped(json, false);
}
function FinishNotifyRequestAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'paymentInfo': (0, FinishNotifyPaymentInfo_1.FinishNotifyPaymentInfoToJSON)(value['paymentInfo']),
        'shopInfo': (0, ShopInfo_1.ShopInfoToJSON)(value['shopInfo']),
        'extendInfo': value['extendInfo'],
        'extendInfo.closedReason': value['extendInfoClosedReason'],
        'paidTime': value['paidTime'],
    };
}
const propertyValidationAttributesMap = {
    extendInfo: {
        maxLength: 4096,
    },
    extendInfoClosedReason: {
        maxLength: 64,
    },
    paidTime: {
        maxLength: 25,
        pattern: new RegExp('/^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+07:00$/'.slice(1, -1)),
    },
};
function validateFinishNotifyRequestAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, FinishNotifyPaymentInfo_1.validateFinishNotifyPaymentInfo)(value.paymentInfo));
    validationErrorContexts.push(...(0, ShopInfo_1.validateShopInfo)(value.shopInfo));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfoClosedReason', value.extendInfoClosedReason, propertyValidationAttributesMap['extendInfoClosedReason']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('paidTime', value.paidTime, propertyValidationAttributesMap['paidTime']));
    return validationErrorContexts;
}
