// dana-node/dist/webhook/v1/FinishNotifyResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFinishNotifyResponse = instanceOfFinishNotifyResponse;
exports.FinishNotifyResponseFromJSON = FinishNotifyResponseFromJSON;
exports.FinishNotifyResponseFromJSONTyped = FinishNotifyResponseFromJSONTyped;
exports.FinishNotifyResponseToJSON = FinishNotifyResponseToJSON;
exports.FinishNotifyResponseToJSONTyped = FinishNotifyResponseToJSONTyped;
exports.validateFinishNotifyResponse = validateFinishNotifyResponse;
const runtime_1 = require("../../runtime");
/**
 * Check if a given object implements the FinishNotifyResponse interface.
 */
function instanceOfFinishNotifyResponse(value) {
    if (!('responseCode' in value) || value['responseCode'] === undefined)
        return false;
    if (!('responseMessage' in value) || value['responseMessage'] === undefined)
        return false;
    return true;
}
function FinishNotifyResponseFromJSON(json) {
    return FinishNotifyResponseFromJSONTyped(json, false);
}
function FinishNotifyResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'],
        'responseMessage': json['responseMessage'],
    };
}
function FinishNotifyResponseToJSON(json) {
    return FinishNotifyResponseToJSONTyped(json, false);
}
function FinishNotifyResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
};
function validateFinishNotifyResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    return validationErrorContexts;
}
