// dana-node/dist/webhook/v1/FinishNotifyPaymentInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFinishNotifyPaymentInfo = instanceOfFinishNotifyPaymentInfo;
exports.FinishNotifyPaymentInfoFromJSON = FinishNotifyPaymentInfoFromJSON;
exports.FinishNotifyPaymentInfoFromJSONTyped = FinishNotifyPaymentInfoFromJSONTyped;
exports.FinishNotifyPaymentInfoToJSON = FinishNotifyPaymentInfoToJSON;
exports.FinishNotifyPaymentInfoToJSONTyped = FinishNotifyPaymentInfoToJSONTyped;
exports.validateFinishNotifyPaymentInfo = validateFinishNotifyPaymentInfo;
const runtime_1 = require("../../runtime");
const PayOptionInfo_1 = require("./PayOptionInfo");
/**
 * Check if a given object implements the FinishNotifyPaymentInfo interface.
 */
function instanceOfFinishNotifyPaymentInfo(value) {
    if (!('payOptionInfos' in value) || value['payOptionInfos'] === undefined)
        return false;
    return true;
}
function FinishNotifyPaymentInfoFromJSON(json) {
    return FinishNotifyPaymentInfoFromJSONTyped(json, false);
}
function FinishNotifyPaymentInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'cashierRequestId': json['cashierRequestId'] == null ? undefined : json['cashierRequestId'],
        'paidTime': json['paidTime'] == null ? undefined : json['paidTime'],
        'payOptionInfos': json['payOptionInfos'] ? (json['payOptionInfos'].map(PayOptionInfo_1.PayOptionInfoFromJSON)) : [],
        'payRequestExtendInfo': json['payRequestExtendInfo'] == null ? undefined : json['payRequestExtendInfo'],
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
    };
}
function FinishNotifyPaymentInfoToJSON(json) {
    return FinishNotifyPaymentInfoToJSONTyped(json, false);
}
function FinishNotifyPaymentInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'cashierRequestId': value['cashierRequestId'],
        'paidTime': value['paidTime'],
        'payOptionInfos': (value['payOptionInfos'].map(PayOptionInfo_1.PayOptionInfoToJSON)),
        'payRequestExtendInfo': value['payRequestExtendInfo'],
        'extendInfo': value['extendInfo'],
    };
}
const propertyValidationAttributesMap = {
    cashierRequestId: {
        maxLength: 64,
    },
    paidTime: {
        maxLength: 25,
        pattern: new RegExp('/^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+07:00$/'.slice(1, -1)),
    },
    payRequestExtendInfo: {
        maxLength: 4096,
    },
    extendInfo: {
        maxLength: 4096,
    },
};
function validateFinishNotifyPaymentInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('cashierRequestId', value.cashierRequestId, propertyValidationAttributesMap['cashierRequestId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('paidTime', value.paidTime, propertyValidationAttributesMap['paidTime']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('payRequestExtendInfo', value.payRequestExtendInfo, propertyValidationAttributesMap['payRequestExtendInfo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    return validationErrorContexts;
}
