// dana-node/dist/disbursement/v1/models/BankAccountInquiryRequestAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BankAccountInquiryRequestAdditionalInfoChargeTargetEnum = void 0;
exports.instanceOfBankAccountInquiryRequestAdditionalInfo = instanceOfBankAccountInquiryRequestAdditionalInfo;
exports.BankAccountInquiryRequestAdditionalInfoFromJSON = BankAccountInquiryRequestAdditionalInfoFromJSON;
exports.BankAccountInquiryRequestAdditionalInfoFromJSONTyped = BankAccountInquiryRequestAdditionalInfoFromJSONTyped;
exports.BankAccountInquiryRequestAdditionalInfoToJSON = BankAccountInquiryRequestAdditionalInfoToJSON;
exports.BankAccountInquiryRequestAdditionalInfoToJSONTyped = BankAccountInquiryRequestAdditionalInfoToJSONTyped;
exports.validateBankAccountInquiryRequestAdditionalInfo = validateBankAccountInquiryRequestAdditionalInfo;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.BankAccountInquiryRequestAdditionalInfoChargeTargetEnum = {
    Division: 'DIVISION',
    Merchant: 'MERCHANT'
};
/**
 * Check if a given object implements the BankAccountInquiryRequestAdditionalInfo interface.
 */
function instanceOfBankAccountInquiryRequestAdditionalInfo(value) {
    if (!('fundType' in value) || value['fundType'] === undefined)
        return false;
    if (!('beneficiaryBankCode' in value) || value['beneficiaryBankCode'] === undefined)
        return false;
    return true;
}
function BankAccountInquiryRequestAdditionalInfoFromJSON(json) {
    return BankAccountInquiryRequestAdditionalInfoFromJSONTyped(json, false);
}
function BankAccountInquiryRequestAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'fundType': json['fundType'],
        'externalDivisionId': json['externalDivisionId'] == null ? undefined : json['externalDivisionId'],
        'chargeTarget': json['chargeTarget'] == null ? undefined : json['chargeTarget'],
        'beneficiaryBankCode': json['beneficiaryBankCode'],
        'beneficiaryAccountName': json['beneficiaryAccountName'] == null ? undefined : json['beneficiaryAccountName'],
        'accountType': json['accountType'] == null ? undefined : json['accountType'],
        'accessToken': json['accessToken'] == null ? undefined : json['accessToken'],
    };
}
function BankAccountInquiryRequestAdditionalInfoToJSON(json) {
    return BankAccountInquiryRequestAdditionalInfoToJSONTyped(json, false);
}
function BankAccountInquiryRequestAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'fundType': value['fundType'],
        'externalDivisionId': value['externalDivisionId'],
        'chargeTarget': value['chargeTarget'],
        'beneficiaryBankCode': value['beneficiaryBankCode'],
        'beneficiaryAccountName': value['beneficiaryAccountName'],
        'accountType': value['accountType'],
        'accessToken': value['accessToken'],
    };
}
const propertyValidationAttributesMap = {
    fundType: {
        maxLength: 64,
    },
    externalDivisionId: {
        maxLength: 64,
    },
    chargeTarget: {
        maxLength: 64,
    },
    beneficiaryBankCode: {
        maxLength: 8,
    },
    beneficiaryAccountName: {
        maxLength: 64,
    },
    accountType: {
        maxLength: 64,
    },
    accessToken: {
        maxLength: 512,
    },
};
function validateBankAccountInquiryRequestAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('fundType', value.fundType, propertyValidationAttributesMap['fundType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('externalDivisionId', value.externalDivisionId, propertyValidationAttributesMap['externalDivisionId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('chargeTarget', value.chargeTarget, propertyValidationAttributesMap['chargeTarget']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('beneficiaryBankCode', value.beneficiaryBankCode, propertyValidationAttributesMap['beneficiaryBankCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('beneficiaryAccountName', value.beneficiaryAccountName, propertyValidationAttributesMap['beneficiaryAccountName']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('accountType', value.accountType, propertyValidationAttributesMap['accountType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('accessToken', value.accessToken, propertyValidationAttributesMap['accessToken']));
    return validationErrorContexts;
}
