// dana-node/dist/disbursement/v1/models/DanaAccountInquiryRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfDanaAccountInquiryRequest = instanceOfDanaAccountInquiryRequest;
exports.DanaAccountInquiryRequestFromJSON = DanaAccountInquiryRequestFromJSON;
exports.DanaAccountInquiryRequestFromJSONTyped = DanaAccountInquiryRequestFromJSONTyped;
exports.DanaAccountInquiryRequestToJSON = DanaAccountInquiryRequestToJSON;
exports.DanaAccountInquiryRequestToJSONTyped = DanaAccountInquiryRequestToJSONTyped;
exports.validateDanaAccountInquiryRequest = validateDanaAccountInquiryRequest;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
const DanaAccountInquiryRequestAdditionalInfo_1 = require("./DanaAccountInquiryRequestAdditionalInfo");
/**
 * Check if a given object implements the DanaAccountInquiryRequest interface.
 */
function instanceOfDanaAccountInquiryRequest(value) {
    if (!('amount' in value) || value['amount'] === undefined)
        return false;
    if (!('additionalInfo' in value) || value['additionalInfo'] === undefined)
        return false;
    return true;
}
function DanaAccountInquiryRequestFromJSON(json) {
    return DanaAccountInquiryRequestFromJSONTyped(json, false);
}
function DanaAccountInquiryRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'partnerReferenceNo': json['partnerReferenceNo'] == null ? undefined : json['partnerReferenceNo'],
        'customerNumber': json['customerNumber'] == null ? undefined : json['customerNumber'],
        'amount': (0, Money_1.MoneyFromJSON)(json['amount']),
        'transactionDate': json['transactionDate'] == null ? undefined : json['transactionDate'],
        'additionalInfo': (0, DanaAccountInquiryRequestAdditionalInfo_1.DanaAccountInquiryRequestAdditionalInfoFromJSON)(json['additionalInfo']),
    };
}
function DanaAccountInquiryRequestToJSON(json) {
    return DanaAccountInquiryRequestToJSONTyped(json, false);
}
function DanaAccountInquiryRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'partnerReferenceNo': value['partnerReferenceNo'],
        'customerNumber': value['customerNumber'],
        'amount': (0, Money_1.MoneyToJSON)(value['amount']),
        'transactionDate': value['transactionDate'],
        'additionalInfo': (0, DanaAccountInquiryRequestAdditionalInfo_1.DanaAccountInquiryRequestAdditionalInfoToJSON)(value['additionalInfo']),
    };
}
const propertyValidationAttributesMap = {
    partnerReferenceNo: {
        maxLength: 64,
    },
    customerNumber: {
        maxLength: 32,
    },
    transactionDate: {
        maxLength: 25,
        pattern: new RegExp('/^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+07:00$/'.slice(1, -1)),
    },
};
function validateDanaAccountInquiryRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('customerNumber', value.customerNumber, propertyValidationAttributesMap['customerNumber']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.amount));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('transactionDate', value.transactionDate, propertyValidationAttributesMap['transactionDate']));
    validationErrorContexts.push(...(0, DanaAccountInquiryRequestAdditionalInfo_1.validateDanaAccountInquiryRequestAdditionalInfo)(value.additionalInfo));
    return validationErrorContexts;
}
