// dana-node/dist/disbursement/v1/models/TransferToDanaResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransferToDanaResponse = instanceOfTransferToDanaResponse;
exports.TransferToDanaResponseFromJSON = TransferToDanaResponseFromJSON;
exports.TransferToDanaResponseFromJSONTyped = TransferToDanaResponseFromJSONTyped;
exports.TransferToDanaResponseToJSON = TransferToDanaResponseToJSON;
exports.TransferToDanaResponseToJSONTyped = TransferToDanaResponseToJSONTyped;
exports.validateTransferToDanaResponse = validateTransferToDanaResponse;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * Check if a given object implements the TransferToDanaResponse interface.
 */
function instanceOfTransferToDanaResponse(value) {
    if (!('responseCode' in value) || value['responseCode'] === undefined)
        return false;
    if (!('responseMessage' in value) || value['responseMessage'] === undefined)
        return false;
    if (!('partnerReferenceNo' in value) || value['partnerReferenceNo'] === undefined)
        return false;
    if (!('amount' in value) || value['amount'] === undefined)
        return false;
    return true;
}
function TransferToDanaResponseFromJSON(json) {
    return TransferToDanaResponseFromJSONTyped(json, false);
}
function TransferToDanaResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'],
        'responseMessage': json['responseMessage'],
        'referenceNo': json['referenceNo'] == null ? undefined : json['referenceNo'],
        'partnerReferenceNo': json['partnerReferenceNo'],
        'sessionId': json['sessionId'] == null ? undefined : json['sessionId'],
        'customerNumber': json['customerNumber'] == null ? undefined : json['customerNumber'],
        'customerName': json['customerName'] == null ? undefined : json['customerName'],
        'amount': (0, Money_1.MoneyFromJSON)(json['amount']),
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function TransferToDanaResponseToJSON(json) {
    return TransferToDanaResponseToJSONTyped(json, false);
}
function TransferToDanaResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
        'referenceNo': value['referenceNo'],
        'partnerReferenceNo': value['partnerReferenceNo'],
        'sessionId': value['sessionId'],
        'customerNumber': value['customerNumber'],
        'customerName': value['customerName'],
        'amount': (0, Money_1.MoneyToJSON)(value['amount']),
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
    referenceNo: {
        maxLength: 64,
    },
    partnerReferenceNo: {
        maxLength: 64,
    },
    sessionId: {
        maxLength: 25,
    },
    customerNumber: {
        maxLength: 32,
    },
    customerName: {
        maxLength: 255,
    },
};
function validateTransferToDanaResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('referenceNo', value.referenceNo, propertyValidationAttributesMap['referenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('sessionId', value.sessionId, propertyValidationAttributesMap['sessionId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('customerNumber', value.customerNumber, propertyValidationAttributesMap['customerNumber']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('customerName', value.customerName, propertyValidationAttributesMap['customerName']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.amount));
    return validationErrorContexts;
}
