// dana-node/dist/disbursement/v1/models/TransferToBankInquiryStatusRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransferToBankInquiryStatusRequest = instanceOfTransferToBankInquiryStatusRequest;
exports.TransferToBankInquiryStatusRequestFromJSON = TransferToBankInquiryStatusRequestFromJSON;
exports.TransferToBankInquiryStatusRequestFromJSONTyped = TransferToBankInquiryStatusRequestFromJSONTyped;
exports.TransferToBankInquiryStatusRequestToJSON = TransferToBankInquiryStatusRequestToJSON;
exports.TransferToBankInquiryStatusRequestToJSONTyped = TransferToBankInquiryStatusRequestToJSONTyped;
exports.validateTransferToBankInquiryStatusRequest = validateTransferToBankInquiryStatusRequest;
const runtime_1 = require("../../../runtime");
/**
 * Check if a given object implements the TransferToBankInquiryStatusRequest interface.
 */
function instanceOfTransferToBankInquiryStatusRequest(value) {
    if (!('serviceCode' in value) || value['serviceCode'] === undefined)
        return false;
    return true;
}
function TransferToBankInquiryStatusRequestFromJSON(json) {
    return TransferToBankInquiryStatusRequestFromJSONTyped(json, false);
}
function TransferToBankInquiryStatusRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'originalPartnerReferenceNo': json['originalPartnerReferenceNo'] == null ? undefined : json['originalPartnerReferenceNo'],
        'originalReferenceNo': json['originalReferenceNo'] == null ? undefined : json['originalReferenceNo'],
        'originalExternalId': json['originalExternalId'] == null ? undefined : json['originalExternalId'],
        'serviceCode': json['serviceCode'],
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function TransferToBankInquiryStatusRequestToJSON(json) {
    return TransferToBankInquiryStatusRequestToJSONTyped(json, false);
}
function TransferToBankInquiryStatusRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'originalPartnerReferenceNo': value['originalPartnerReferenceNo'],
        'originalReferenceNo': value['originalReferenceNo'],
        'originalExternalId': value['originalExternalId'],
        'serviceCode': value['serviceCode'],
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    originalPartnerReferenceNo: {
        maxLength: 64,
    },
    originalReferenceNo: {
        maxLength: 64,
    },
    originalExternalId: {
        maxLength: 36,
    },
    serviceCode: {
        maxLength: 2,
    },
};
function validateTransferToBankInquiryStatusRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalPartnerReferenceNo', value.originalPartnerReferenceNo, propertyValidationAttributesMap['originalPartnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalReferenceNo', value.originalReferenceNo, propertyValidationAttributesMap['originalReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalExternalId', value.originalExternalId, propertyValidationAttributesMap['originalExternalId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('serviceCode', value.serviceCode, propertyValidationAttributesMap['serviceCode']));
    return validationErrorContexts;
}
