// dana-node/dist/disbursement/v1/models/TransferToDanaRequestAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransferToDanaRequestAdditionalInfoChargeTargetEnum = void 0;
exports.instanceOfTransferToDanaRequestAdditionalInfo = instanceOfTransferToDanaRequestAdditionalInfo;
exports.TransferToDanaRequestAdditionalInfoFromJSON = TransferToDanaRequestAdditionalInfoFromJSON;
exports.TransferToDanaRequestAdditionalInfoFromJSONTyped = TransferToDanaRequestAdditionalInfoFromJSONTyped;
exports.TransferToDanaRequestAdditionalInfoToJSON = TransferToDanaRequestAdditionalInfoToJSON;
exports.TransferToDanaRequestAdditionalInfoToJSONTyped = TransferToDanaRequestAdditionalInfoToJSONTyped;
exports.validateTransferToDanaRequestAdditionalInfo = validateTransferToDanaRequestAdditionalInfo;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.TransferToDanaRequestAdditionalInfoChargeTargetEnum = {
    Division: 'DIVISION',
    Merchant: 'MERCHANT'
};
/**
 * Check if a given object implements the TransferToDanaRequestAdditionalInfo interface.
 */
function instanceOfTransferToDanaRequestAdditionalInfo(value) {
    if (!('fundType' in value) || value['fundType'] === undefined)
        return false;
    return true;
}
function TransferToDanaRequestAdditionalInfoFromJSON(json) {
    return TransferToDanaRequestAdditionalInfoFromJSONTyped(json, false);
}
function TransferToDanaRequestAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'accountType': json['accountType'] == null ? undefined : json['accountType'],
        'fundType': json['fundType'],
        'externalDivisionId': json['externalDivisionId'] == null ? undefined : json['externalDivisionId'],
        'chargeTarget': json['chargeTarget'] == null ? undefined : json['chargeTarget'],
        'accessToken': json['accessToken'] == null ? undefined : json['accessToken'],
        'customerId': json['customerId'] == null ? undefined : json['customerId'],
    };
}
function TransferToDanaRequestAdditionalInfoToJSON(json) {
    return TransferToDanaRequestAdditionalInfoToJSONTyped(json, false);
}
function TransferToDanaRequestAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'extendInfo': value['extendInfo'],
        'accountType': value['accountType'],
        'fundType': value['fundType'],
        'externalDivisionId': value['externalDivisionId'],
        'chargeTarget': value['chargeTarget'],
        'accessToken': value['accessToken'],
        'customerId': value['customerId'],
    };
}
const propertyValidationAttributesMap = {
    extendInfo: {
        maxLength: 4096,
    },
    accountType: {
        maxLength: 64,
    },
    fundType: {
        maxLength: 64,
    },
    externalDivisionId: {
        maxLength: 64,
    },
    chargeTarget: {
        maxLength: 64,
    },
    accessToken: {
        maxLength: 512,
    },
    customerId: {
        maxLength: 32,
    },
};
function validateTransferToDanaRequestAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('accountType', value.accountType, propertyValidationAttributesMap['accountType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('fundType', value.fundType, propertyValidationAttributesMap['fundType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('externalDivisionId', value.externalDivisionId, propertyValidationAttributesMap['externalDivisionId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('chargeTarget', value.chargeTarget, propertyValidationAttributesMap['chargeTarget']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('accessToken', value.accessToken, propertyValidationAttributesMap['accessToken']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('customerId', value.customerId, propertyValidationAttributesMap['customerId']));
    return validationErrorContexts;
}
