// dana-node/dist/widget/v1/models/AccountUnbindingRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAccountUnbindingRequest = instanceOfAccountUnbindingRequest;
exports.AccountUnbindingRequestFromJSON = AccountUnbindingRequestFromJSON;
exports.AccountUnbindingRequestFromJSONTyped = AccountUnbindingRequestFromJSONTyped;
exports.AccountUnbindingRequestToJSON = AccountUnbindingRequestToJSON;
exports.AccountUnbindingRequestToJSONTyped = AccountUnbindingRequestToJSONTyped;
exports.validateAccountUnbindingRequest = validateAccountUnbindingRequest;
const runtime_1 = require("../../../runtime");
const AccountUnbindingRequestAdditionalInfo_1 = require("./AccountUnbindingRequestAdditionalInfo");
/**
 * Check if a given object implements the AccountUnbindingRequest interface.
 */
function instanceOfAccountUnbindingRequest(value) {
    if (!('merchantId' in value) || value['merchantId'] === undefined)
        return false;
    if (!('additionalInfo' in value) || value['additionalInfo'] === undefined)
        return false;
    return true;
}
function AccountUnbindingRequestFromJSON(json) {
    return AccountUnbindingRequestFromJSONTyped(json, false);
}
function AccountUnbindingRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'merchantId': json['merchantId'],
        'subMerchantId': json['subMerchantId'] == null ? undefined : json['subMerchantId'],
        'partnerReferenceNo': json['partnerReferenceNo'] == null ? undefined : json['partnerReferenceNo'],
        'linkId': json['linkId'] == null ? undefined : json['linkId'],
        'tokenId': json['tokenId'] == null ? undefined : json['tokenId'],
        'additionalInfo': (0, AccountUnbindingRequestAdditionalInfo_1.AccountUnbindingRequestAdditionalInfoFromJSON)(json['additionalInfo']),
    };
}
function AccountUnbindingRequestToJSON(json) {
    return AccountUnbindingRequestToJSONTyped(json, false);
}
function AccountUnbindingRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'merchantId': value['merchantId'],
        'subMerchantId': value['subMerchantId'],
        'partnerReferenceNo': value['partnerReferenceNo'],
        'linkId': value['linkId'],
        'tokenId': value['tokenId'],
        'additionalInfo': (0, AccountUnbindingRequestAdditionalInfo_1.AccountUnbindingRequestAdditionalInfoToJSON)(value['additionalInfo']),
    };
}
const propertyValidationAttributesMap = {
    merchantId: {
        maxLength: 64,
    },
    subMerchantId: {
        maxLength: 32,
    },
    partnerReferenceNo: {
        maxLength: 64,
    },
    linkId: {
        maxLength: 24,
    },
    tokenId: {
        maxLength: 128,
    },
};
function validateAccountUnbindingRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantId', value.merchantId, propertyValidationAttributesMap['merchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('subMerchantId', value.subMerchantId, propertyValidationAttributesMap['subMerchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('linkId', value.linkId, propertyValidationAttributesMap['linkId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('tokenId', value.tokenId, propertyValidationAttributesMap['tokenId']));
    validationErrorContexts.push(...(0, AccountUnbindingRequestAdditionalInfo_1.validateAccountUnbindingRequestAdditionalInfo)(value.additionalInfo));
    return validationErrorContexts;
}
