// dana-node/dist/widget/v1/models/ApplyTokenBaseRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfApplyTokenBaseRequest = instanceOfApplyTokenBaseRequest;
exports.ApplyTokenBaseRequestFromJSON = ApplyTokenBaseRequestFromJSON;
exports.ApplyTokenBaseRequestFromJSONTyped = ApplyTokenBaseRequestFromJSONTyped;
exports.ApplyTokenBaseRequestToJSON = ApplyTokenBaseRequestToJSON;
exports.ApplyTokenBaseRequestToJSONTyped = ApplyTokenBaseRequestToJSONTyped;
exports.validateApplyTokenBaseRequest = validateApplyTokenBaseRequest;
/**
 * Check if a given object implements the ApplyTokenBaseRequest interface.
 */
function instanceOfApplyTokenBaseRequest(value) {
    return true;
}
function ApplyTokenBaseRequestFromJSON(json) {
    return ApplyTokenBaseRequestFromJSONTyped(json, false);
}
function ApplyTokenBaseRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function ApplyTokenBaseRequestToJSON(json) {
    return ApplyTokenBaseRequestToJSONTyped(json, false);
}
function ApplyTokenBaseRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {};
function validateApplyTokenBaseRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
