// dana-node/dist/widget/v1/models/BalanceInquiryRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBalanceInquiryRequest = instanceOfBalanceInquiryRequest;
exports.BalanceInquiryRequestFromJSON = BalanceInquiryRequestFromJSON;
exports.BalanceInquiryRequestFromJSONTyped = BalanceInquiryRequestFromJSONTyped;
exports.BalanceInquiryRequestToJSON = BalanceInquiryRequestToJSON;
exports.BalanceInquiryRequestToJSONTyped = BalanceInquiryRequestToJSONTyped;
exports.validateBalanceInquiryRequest = validateBalanceInquiryRequest;
const runtime_1 = require("../../../runtime");
const BalanceInquiryRequestAdditionalInfo_1 = require("./BalanceInquiryRequestAdditionalInfo");
/**
 * Check if a given object implements the BalanceInquiryRequest interface.
 */
function instanceOfBalanceInquiryRequest(value) {
    if (!('additionalInfo' in value) || value['additionalInfo'] === undefined)
        return false;
    return true;
}
function BalanceInquiryRequestFromJSON(json) {
    return BalanceInquiryRequestFromJSONTyped(json, false);
}
function BalanceInquiryRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'partnerReferenceNo': json['partnerReferenceNo'] == null ? undefined : json['partnerReferenceNo'],
        'balanceTypes': json['balanceTypes'] == null ? undefined : json['balanceTypes'],
        'additionalInfo': (0, BalanceInquiryRequestAdditionalInfo_1.BalanceInquiryRequestAdditionalInfoFromJSON)(json['additionalInfo']),
    };
}
function BalanceInquiryRequestToJSON(json) {
    return BalanceInquiryRequestToJSONTyped(json, false);
}
function BalanceInquiryRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'partnerReferenceNo': value['partnerReferenceNo'],
        'balanceTypes': value['balanceTypes'],
        'additionalInfo': (0, BalanceInquiryRequestAdditionalInfo_1.BalanceInquiryRequestAdditionalInfoToJSON)(value['additionalInfo']),
    };
}
const propertyValidationAttributesMap = {
    partnerReferenceNo: {
        maxLength: 64,
    },
};
function validateBalanceInquiryRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...(0, BalanceInquiryRequestAdditionalInfo_1.validateBalanceInquiryRequestAdditionalInfo)(value.additionalInfo));
    return validationErrorContexts;
}
