// dana-node/dist/widget/v1/models/BalanceInquiryResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBalanceInquiryResponse = instanceOfBalanceInquiryResponse;
exports.BalanceInquiryResponseFromJSON = BalanceInquiryResponseFromJSON;
exports.BalanceInquiryResponseFromJSONTyped = BalanceInquiryResponseFromJSONTyped;
exports.BalanceInquiryResponseToJSON = BalanceInquiryResponseToJSON;
exports.BalanceInquiryResponseToJSONTyped = BalanceInquiryResponseToJSONTyped;
exports.validateBalanceInquiryResponse = validateBalanceInquiryResponse;
const runtime_1 = require("../../../runtime");
const AccountInfo_1 = require("./AccountInfo");
/**
 * Check if a given object implements the BalanceInquiryResponse interface.
 */
function instanceOfBalanceInquiryResponse(value) {
    if (!('responseCode' in value) || value['responseCode'] === undefined)
        return false;
    if (!('responseMessage' in value) || value['responseMessage'] === undefined)
        return false;
    return true;
}
function BalanceInquiryResponseFromJSON(json) {
    return BalanceInquiryResponseFromJSONTyped(json, false);
}
function BalanceInquiryResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'],
        'responseMessage': json['responseMessage'],
        'referenceNo': json['referenceNo'] == null ? undefined : json['referenceNo'],
        'partnerReferenceNo': json['partnerReferenceNo'] == null ? undefined : json['partnerReferenceNo'],
        'name': json['name'] == null ? undefined : json['name'],
        'accountInfos': json['accountInfos'] == null ? undefined : (json['accountInfos'].map(AccountInfo_1.AccountInfoFromJSON)),
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function BalanceInquiryResponseToJSON(json) {
    return BalanceInquiryResponseToJSONTyped(json, false);
}
function BalanceInquiryResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
        'referenceNo': value['referenceNo'],
        'partnerReferenceNo': value['partnerReferenceNo'],
        'name': value['name'],
        'accountInfos': value['accountInfos'] == null ? undefined : (value['accountInfos'].map(AccountInfo_1.AccountInfoToJSON)),
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
    referenceNo: {
        maxLength: 64,
    },
    partnerReferenceNo: {
        maxLength: 64,
    },
    name: {
        maxLength: 140,
    },
};
function validateBalanceInquiryResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('referenceNo', value.referenceNo, propertyValidationAttributesMap['referenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('name', value.name, propertyValidationAttributesMap['name']));
    return validationErrorContexts;
}
