// dana-node/dist/widget/v1/models/CancelOrderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCancelOrderRequest = instanceOfCancelOrderRequest;
exports.CancelOrderRequestFromJSON = CancelOrderRequestFromJSON;
exports.CancelOrderRequestFromJSONTyped = CancelOrderRequestFromJSONTyped;
exports.CancelOrderRequestToJSON = CancelOrderRequestToJSON;
exports.CancelOrderRequestToJSONTyped = CancelOrderRequestToJSONTyped;
exports.validateCancelOrderRequest = validateCancelOrderRequest;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * Check if a given object implements the CancelOrderRequest interface.
 */
function instanceOfCancelOrderRequest(value) {
    if (!('originalPartnerReferenceNo' in value) || value['originalPartnerReferenceNo'] === undefined)
        return false;
    if (!('merchantId' in value) || value['merchantId'] === undefined)
        return false;
    return true;
}
function CancelOrderRequestFromJSON(json) {
    return CancelOrderRequestFromJSONTyped(json, false);
}
function CancelOrderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'originalPartnerReferenceNo': json['originalPartnerReferenceNo'],
        'originalReferenceNo': json['originalReferenceNo'] == null ? undefined : json['originalReferenceNo'],
        'originalExternalId': json['originalExternalId'] == null ? undefined : json['originalExternalId'],
        'merchantId': json['merchantId'],
        'subMerchantId': json['subMerchantId'] == null ? undefined : json['subMerchantId'],
        'reason': json['reason'] == null ? undefined : json['reason'],
        'externalStoreId': json['externalStoreId'] == null ? undefined : json['externalStoreId'],
        'amount': json['amount'] == null ? undefined : (0, Money_1.MoneyFromJSON)(json['amount']),
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function CancelOrderRequestToJSON(json) {
    return CancelOrderRequestToJSONTyped(json, false);
}
function CancelOrderRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'originalPartnerReferenceNo': value['originalPartnerReferenceNo'],
        'originalReferenceNo': value['originalReferenceNo'],
        'originalExternalId': value['originalExternalId'],
        'merchantId': value['merchantId'],
        'subMerchantId': value['subMerchantId'],
        'reason': value['reason'],
        'externalStoreId': value['externalStoreId'],
        'amount': (0, Money_1.MoneyToJSON)(value['amount']),
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    originalPartnerReferenceNo: {
        maxLength: 64,
    },
    originalReferenceNo: {
        maxLength: 64,
    },
    originalExternalId: {
        maxLength: 36,
    },
    merchantId: {
        maxLength: 64,
    },
    subMerchantId: {
        maxLength: 32,
    },
    reason: {
        maxLength: 256,
    },
    externalStoreId: {
        maxLength: 64,
    },
};
function validateCancelOrderRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalPartnerReferenceNo', value.originalPartnerReferenceNo, propertyValidationAttributesMap['originalPartnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalReferenceNo', value.originalReferenceNo, propertyValidationAttributesMap['originalReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalExternalId', value.originalExternalId, propertyValidationAttributesMap['originalExternalId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantId', value.merchantId, propertyValidationAttributesMap['merchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('subMerchantId', value.subMerchantId, propertyValidationAttributesMap['subMerchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('reason', value.reason, propertyValidationAttributesMap['reason']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('externalStoreId', value.externalStoreId, propertyValidationAttributesMap['externalStoreId']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.amount));
    return validationErrorContexts;
}
