// dana-node/dist/widget/v1/models/Goods.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfGoods = instanceOfGoods;
exports.GoodsFromJSON = GoodsFromJSON;
exports.GoodsFromJSONTyped = GoodsFromJSONTyped;
exports.GoodsToJSON = GoodsToJSON;
exports.GoodsToJSONTyped = GoodsToJSONTyped;
exports.validateGoods = validateGoods;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * Check if a given object implements the Goods interface.
 */
function instanceOfGoods(value) {
    if (!('name' in value) || value['name'] === undefined)
        return false;
    if (!('merchantGoodsId' in value) || value['merchantGoodsId'] === undefined)
        return false;
    if (!('description' in value) || value['description'] === undefined)
        return false;
    if (!('category' in value) || value['category'] === undefined)
        return false;
    if (!('price' in value) || value['price'] === undefined)
        return false;
    if (!('quantity' in value) || value['quantity'] === undefined)
        return false;
    return true;
}
function GoodsFromJSON(json) {
    return GoodsFromJSONTyped(json, false);
}
function GoodsFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'name': json['name'],
        'merchantGoodsId': json['merchantGoodsId'],
        'description': json['description'],
        'category': json['category'],
        'price': (0, Money_1.MoneyFromJSON)(json['price']),
        'unit': json['unit'] == null ? undefined : json['unit'],
        'quantity': json['quantity'],
        'merchantShippingId': json['merchantShippingId'] == null ? undefined : json['merchantShippingId'],
        'snapshotUrl': json['snapshotUrl'] == null ? undefined : json['snapshotUrl'],
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
    };
}
function GoodsToJSON(json) {
    return GoodsToJSONTyped(json, false);
}
function GoodsToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'name': value['name'],
        'merchantGoodsId': value['merchantGoodsId'],
        'description': value['description'],
        'category': value['category'],
        'price': (0, Money_1.MoneyToJSON)(value['price']),
        'unit': value['unit'],
        'quantity': value['quantity'],
        'merchantShippingId': value['merchantShippingId'],
        'snapshotUrl': value['snapshotUrl'],
        'extendInfo': value['extendInfo'],
    };
}
const propertyValidationAttributesMap = {
    name: {
        maxLength: 64,
    },
    merchantGoodsId: {
        maxLength: 64,
    },
    description: {
        maxLength: 1024,
    },
    category: {
        maxLength: 64,
    },
    unit: {
        maxLength: 64,
    },
    quantity: {
        maxLength: 16,
    },
    merchantShippingId: {
        maxLength: 64,
    },
    snapshotUrl: {
        maxLength: 512,
    },
    extendInfo: {
        maxLength: 4096,
    },
};
function validateGoods(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('name', value.name, propertyValidationAttributesMap['name']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantGoodsId', value.merchantGoodsId, propertyValidationAttributesMap['merchantGoodsId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('description', value.description, propertyValidationAttributesMap['description']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('category', value.category, propertyValidationAttributesMap['category']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.price));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('unit', value.unit, propertyValidationAttributesMap['unit']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('quantity', value.quantity, propertyValidationAttributesMap['quantity']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantShippingId', value.merchantShippingId, propertyValidationAttributesMap['merchantShippingId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('snapshotUrl', value.snapshotUrl, propertyValidationAttributesMap['snapshotUrl']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    return validationErrorContexts;
}
