// dana-node/dist/widget/v1/models/InternationalOrderInfoExchangeRate.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfInternationalOrderInfoExchangeRate = instanceOfInternationalOrderInfoExchangeRate;
exports.InternationalOrderInfoExchangeRateFromJSON = InternationalOrderInfoExchangeRateFromJSON;
exports.InternationalOrderInfoExchangeRateFromJSONTyped = InternationalOrderInfoExchangeRateFromJSONTyped;
exports.InternationalOrderInfoExchangeRateToJSON = InternationalOrderInfoExchangeRateToJSON;
exports.InternationalOrderInfoExchangeRateToJSONTyped = InternationalOrderInfoExchangeRateToJSONTyped;
exports.validateInternationalOrderInfoExchangeRate = validateInternationalOrderInfoExchangeRate;
/**
 * Check if a given object implements the InternationalOrderInfoExchangeRate interface.
 */
function instanceOfInternationalOrderInfoExchangeRate(value) {
    return true;
}
function InternationalOrderInfoExchangeRateFromJSON(json) {
    return InternationalOrderInfoExchangeRateFromJSONTyped(json, false);
}
function InternationalOrderInfoExchangeRateFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'rate': json['rate'] == null ? undefined : json['rate'],
        'exchangeRelation': json['exchangeRelation'] == null ? undefined : json['exchangeRelation'],
    };
}
function InternationalOrderInfoExchangeRateToJSON(json) {
    return InternationalOrderInfoExchangeRateToJSONTyped(json, false);
}
function InternationalOrderInfoExchangeRateToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'rate': value['rate'],
        'exchangeRelation': value['exchangeRelation'],
    };
}
const propertyValidationAttributesMap = {};
function validateInternationalOrderInfoExchangeRate(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
