// dana-node/dist/widget/v1/models/PaymentPromoInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentPromoInfoPromoTypeEnum = void 0;
exports.instanceOfPaymentPromoInfo = instanceOfPaymentPromoInfo;
exports.PaymentPromoInfoFromJSON = PaymentPromoInfoFromJSON;
exports.PaymentPromoInfoFromJSONTyped = PaymentPromoInfoFromJSONTyped;
exports.PaymentPromoInfoToJSON = PaymentPromoInfoToJSON;
exports.PaymentPromoInfoToJSONTyped = PaymentPromoInfoToJSONTyped;
exports.validatePaymentPromoInfo = validatePaymentPromoInfo;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * @export
 */
exports.PaymentPromoInfoPromoTypeEnum = {
    CashBack: 'CASH_BACK',
    Discount: 'DISCOUNT',
    Voucher: 'VOUCHER',
    Point: 'POINT'
};
/**
 * Check if a given object implements the PaymentPromoInfo interface.
 */
function instanceOfPaymentPromoInfo(value) {
    if (!('promoId' in value) || value['promoId'] === undefined)
        return false;
    if (!('promoName' in value) || value['promoName'] === undefined)
        return false;
    if (!('promoType' in value) || value['promoType'] === undefined)
        return false;
    if (!('savingsAmount' in value) || value['savingsAmount'] === undefined)
        return false;
    return true;
}
function PaymentPromoInfoFromJSON(json) {
    return PaymentPromoInfoFromJSONTyped(json, false);
}
function PaymentPromoInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'promoId': json['promoId'],
        'promoName': json['promoName'],
        'promoType': json['promoType'],
        'savingsAmount': (0, Money_1.MoneyFromJSON)(json['savingsAmount']),
    };
}
function PaymentPromoInfoToJSON(json) {
    return PaymentPromoInfoToJSONTyped(json, false);
}
function PaymentPromoInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'promoId': value['promoId'],
        'promoName': value['promoName'],
        'promoType': value['promoType'],
        'savingsAmount': (0, Money_1.MoneyToJSON)(value['savingsAmount']),
    };
}
const propertyValidationAttributesMap = {
    promoId: {
        maxLength: 128,
    },
    promoName: {
        maxLength: 128,
    },
};
function validatePaymentPromoInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('promoId', value.promoId, propertyValidationAttributesMap['promoId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('promoName', value.promoName, propertyValidationAttributesMap['promoName']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.savingsAmount));
    return validationErrorContexts;
}
