// dana-node/dist/widget/v1/models/PayOptionDetail.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPayOptionDetail = instanceOfPayOptionDetail;
exports.PayOptionDetailFromJSON = PayOptionDetailFromJSON;
exports.PayOptionDetailFromJSONTyped = PayOptionDetailFromJSONTyped;
exports.PayOptionDetailToJSON = PayOptionDetailToJSON;
exports.PayOptionDetailToJSONTyped = PayOptionDetailToJSONTyped;
exports.validatePayOptionDetail = validatePayOptionDetail;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
const PayOptionDetailAdditionalInfo_1 = require("./PayOptionDetailAdditionalInfo");
/**
 * Check if a given object implements the PayOptionDetail interface.
 */
function instanceOfPayOptionDetail(value) {
    if (!('payMethod' in value) || value['payMethod'] === undefined)
        return false;
    if (!('payOption' in value) || value['payOption'] === undefined)
        return false;
    return true;
}
function PayOptionDetailFromJSON(json) {
    return PayOptionDetailFromJSONTyped(json, false);
}
function PayOptionDetailFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'payMethod': json['payMethod'],
        'payOption': json['payOption'],
        'transAmount': json['transAmount'] == null ? undefined : (0, Money_1.MoneyFromJSON)(json['transAmount']),
        'feeAmount': json['feeAmount'] == null ? undefined : (0, Money_1.MoneyFromJSON)(json['feeAmount']),
        'cardToken': json['cardToken'] == null ? undefined : json['cardToken'],
        'merchantToken': json['merchantToken'] == null ? undefined : json['merchantToken'],
        'additionalInfo': json['additionalInfo'] == null ? undefined : (0, PayOptionDetailAdditionalInfo_1.PayOptionDetailAdditionalInfoFromJSON)(json['additionalInfo']),
    };
}
function PayOptionDetailToJSON(json) {
    return PayOptionDetailToJSONTyped(json, false);
}
function PayOptionDetailToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'payMethod': value['payMethod'],
        'payOption': value['payOption'],
        'transAmount': (0, Money_1.MoneyToJSON)(value['transAmount']),
        'feeAmount': (0, Money_1.MoneyToJSON)(value['feeAmount']),
        'cardToken': value['cardToken'],
        'merchantToken': value['merchantToken'],
        'additionalInfo': (0, PayOptionDetailAdditionalInfo_1.PayOptionDetailAdditionalInfoToJSON)(value['additionalInfo']),
    };
}
const propertyValidationAttributesMap = {
    payMethod: {
        maxLength: 64,
    },
    payOption: {
        maxLength: 64,
    },
    cardToken: {
        maxLength: 64,
    },
    merchantToken: {
        maxLength: 64,
    },
};
function validatePayOptionDetail(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('payMethod', value.payMethod, propertyValidationAttributesMap['payMethod']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('payOption', value.payOption, propertyValidationAttributesMap['payOption']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.transAmount));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.feeAmount));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('cardToken', value.cardToken, propertyValidationAttributesMap['cardToken']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantToken', value.merchantToken, propertyValidationAttributesMap['merchantToken']));
    validationErrorContexts.push(...(0, PayOptionDetailAdditionalInfo_1.validatePayOptionDetailAdditionalInfo)(value.additionalInfo));
    return validationErrorContexts;
}
