// dana-node/dist/widget/v1/models/QueryUserProfileRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryUserProfileRequestUserResourcesEnum = void 0;
exports.instanceOfQueryUserProfileRequest = instanceOfQueryUserProfileRequest;
exports.QueryUserProfileRequestFromJSON = QueryUserProfileRequestFromJSON;
exports.QueryUserProfileRequestFromJSONTyped = QueryUserProfileRequestFromJSONTyped;
exports.QueryUserProfileRequestToJSON = QueryUserProfileRequestToJSON;
exports.QueryUserProfileRequestToJSONTyped = QueryUserProfileRequestToJSONTyped;
exports.validateQueryUserProfileRequest = validateQueryUserProfileRequest;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.QueryUserProfileRequestUserResourcesEnum = {
    Balance: 'BALANCE',
    TransactionUrl: 'TRANSACTION_URL',
    MaskDanaId: 'MASK_DANA_ID',
    TopupUrl: 'TOPUP_URL',
    Ott: 'OTT',
    UserKyc: 'USER_KYC'
};
/**
 * Check if a given object implements the QueryUserProfileRequest interface.
 */
function instanceOfQueryUserProfileRequest(value) {
    if (!('userResources' in value) || value['userResources'] === undefined)
        return false;
    return true;
}
function QueryUserProfileRequestFromJSON(json) {
    return QueryUserProfileRequestFromJSONTyped(json, false);
}
function QueryUserProfileRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'userResources': json['userResources'],
    };
}
function QueryUserProfileRequestToJSON(json) {
    return QueryUserProfileRequestToJSONTyped(json, false);
}
function QueryUserProfileRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'userResources': value['userResources'],
    };
}
const propertyValidationAttributesMap = {
    userResources: {
        minItems: 1,
        uniqueItems: false,
    },
};
function validateQueryUserProfileRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('userResources', value.userResources, propertyValidationAttributesMap['userResources']));
    return validationErrorContexts;
}
