// dana-node/dist/widget/v1/models/QueryUserProfileResponseResponseBody.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryUserProfileResponseResponseBody = instanceOfQueryUserProfileResponseResponseBody;
exports.QueryUserProfileResponseResponseBodyFromJSON = QueryUserProfileResponseResponseBodyFromJSON;
exports.QueryUserProfileResponseResponseBodyFromJSONTyped = QueryUserProfileResponseResponseBodyFromJSONTyped;
exports.QueryUserProfileResponseResponseBodyToJSON = QueryUserProfileResponseResponseBodyToJSON;
exports.QueryUserProfileResponseResponseBodyToJSONTyped = QueryUserProfileResponseResponseBodyToJSONTyped;
exports.validateQueryUserProfileResponseResponseBody = validateQueryUserProfileResponseResponseBody;
const UserResourceInfo_1 = require("./UserResourceInfo");
const ResultInfo_1 = require("./ResultInfo");
/**
 * Check if a given object implements the QueryUserProfileResponseResponseBody interface.
 */
function instanceOfQueryUserProfileResponseResponseBody(value) {
    if (!('resultInfo' in value) || value['resultInfo'] === undefined)
        return false;
    return true;
}
function QueryUserProfileResponseResponseBodyFromJSON(json) {
    return QueryUserProfileResponseResponseBodyFromJSONTyped(json, false);
}
function QueryUserProfileResponseResponseBodyFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'resultInfo': (0, ResultInfo_1.ResultInfoFromJSON)(json['resultInfo']),
        'userResourceInfos': json['userResourceInfos'] == null ? undefined : (json['userResourceInfos'].map(UserResourceInfo_1.UserResourceInfoFromJSON)),
    };
}
function QueryUserProfileResponseResponseBodyToJSON(json) {
    return QueryUserProfileResponseResponseBodyToJSONTyped(json, false);
}
function QueryUserProfileResponseResponseBodyToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'resultInfo': (0, ResultInfo_1.ResultInfoToJSON)(value['resultInfo']),
        'userResourceInfos': value['userResourceInfos'] == null ? undefined : (value['userResourceInfos'].map(UserResourceInfo_1.UserResourceInfoToJSON)),
    };
}
const propertyValidationAttributesMap = {};
function validateQueryUserProfileResponseResponseBody(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, ResultInfo_1.validateResultInfo)(value.resultInfo));
    return validationErrorContexts;
}
