// dana-node/dist/widget/v1/models/QueryUserProfileResponseResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryUserProfileResponseResponse = instanceOfQueryUserProfileResponseResponse;
exports.QueryUserProfileResponseResponseFromJSON = QueryUserProfileResponseResponseFromJSON;
exports.QueryUserProfileResponseResponseFromJSONTyped = QueryUserProfileResponseResponseFromJSONTyped;
exports.QueryUserProfileResponseResponseToJSON = QueryUserProfileResponseResponseToJSON;
exports.QueryUserProfileResponseResponseToJSONTyped = QueryUserProfileResponseResponseToJSONTyped;
exports.validateQueryUserProfileResponseResponse = validateQueryUserProfileResponseResponse;
const QueryUserProfileResponseResponseHead_1 = require("./QueryUserProfileResponseResponseHead");
const QueryUserProfileResponseResponseBody_1 = require("./QueryUserProfileResponseResponseBody");
/**
 * Check if a given object implements the QueryUserProfileResponseResponse interface.
 */
function instanceOfQueryUserProfileResponseResponse(value) {
    if (!('head' in value) || value['head'] === undefined)
        return false;
    if (!('body' in value) || value['body'] === undefined)
        return false;
    return true;
}
function QueryUserProfileResponseResponseFromJSON(json) {
    return QueryUserProfileResponseResponseFromJSONTyped(json, false);
}
function QueryUserProfileResponseResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'head': (0, QueryUserProfileResponseResponseHead_1.QueryUserProfileResponseResponseHeadFromJSON)(json['head']),
        'body': (0, QueryUserProfileResponseResponseBody_1.QueryUserProfileResponseResponseBodyFromJSON)(json['body']),
    };
}
function QueryUserProfileResponseResponseToJSON(json) {
    return QueryUserProfileResponseResponseToJSONTyped(json, false);
}
function QueryUserProfileResponseResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'head': (0, QueryUserProfileResponseResponseHead_1.QueryUserProfileResponseResponseHeadToJSON)(value['head']),
        'body': (0, QueryUserProfileResponseResponseBody_1.QueryUserProfileResponseResponseBodyToJSON)(value['body']),
    };
}
const propertyValidationAttributesMap = {};
function validateQueryUserProfileResponseResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, QueryUserProfileResponseResponseHead_1.validateQueryUserProfileResponseResponseHead)(value.head));
    validationErrorContexts.push(...(0, QueryUserProfileResponseResponseBody_1.validateQueryUserProfileResponseResponseBody)(value.body));
    return validationErrorContexts;
}
