// dana-node/dist/widget/v1/models/QueryUserProfileResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryUserProfileResponse = instanceOfQueryUserProfileResponse;
exports.QueryUserProfileResponseFromJSON = QueryUserProfileResponseFromJSON;
exports.QueryUserProfileResponseFromJSONTyped = QueryUserProfileResponseFromJSONTyped;
exports.QueryUserProfileResponseToJSON = QueryUserProfileResponseToJSON;
exports.QueryUserProfileResponseToJSONTyped = QueryUserProfileResponseToJSONTyped;
exports.validateQueryUserProfileResponse = validateQueryUserProfileResponse;
const QueryUserProfileResponseResponse_1 = require("./QueryUserProfileResponseResponse");
/**
 * Check if a given object implements the QueryUserProfileResponse interface.
 */
function instanceOfQueryUserProfileResponse(value) {
    if (!('response' in value) || value['response'] === undefined)
        return false;
    return true;
}
function QueryUserProfileResponseFromJSON(json) {
    return QueryUserProfileResponseFromJSONTyped(json, false);
}
function QueryUserProfileResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'response': (0, QueryUserProfileResponseResponse_1.QueryUserProfileResponseResponseFromJSON)(json['response']),
        'signature': json['signature'] == null ? undefined : json['signature'],
    };
}
function QueryUserProfileResponseToJSON(json) {
    return QueryUserProfileResponseToJSONTyped(json, false);
}
function QueryUserProfileResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'response': (0, QueryUserProfileResponseResponse_1.QueryUserProfileResponseResponseToJSON)(value['response']),
        'signature': value['signature'],
    };
}
const propertyValidationAttributesMap = {};
function validateQueryUserProfileResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, QueryUserProfileResponseResponse_1.validateQueryUserProfileResponseResponse)(value.response));
    return validationErrorContexts;
}
