// dana-node/dist/widget/v1/models/UserResourceInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserResourceInfoResourceTypeEnum = void 0;
exports.instanceOfUserResourceInfo = instanceOfUserResourceInfo;
exports.UserResourceInfoFromJSON = UserResourceInfoFromJSON;
exports.UserResourceInfoFromJSONTyped = UserResourceInfoFromJSONTyped;
exports.UserResourceInfoToJSON = UserResourceInfoToJSON;
exports.UserResourceInfoToJSONTyped = UserResourceInfoToJSONTyped;
exports.validateUserResourceInfo = validateUserResourceInfo;
/**
 * @export
 */
exports.UserResourceInfoResourceTypeEnum = {
    Balance: 'BALANCE',
    TransactionUrl: 'TRANSACTION_URL',
    MaskDanaId: 'MASK_DANA_ID',
    TopupUrl: 'TOPUP_URL',
    Ott: 'OTT',
    UserKyc: 'USER_KYC'
};
/**
 * Check if a given object implements the UserResourceInfo interface.
 */
function instanceOfUserResourceInfo(value) {
    if (!('resourceType' in value) || value['resourceType'] === undefined)
        return false;
    if (!('value' in value) || value['value'] === undefined)
        return false;
    return true;
}
function UserResourceInfoFromJSON(json) {
    return UserResourceInfoFromJSONTyped(json, false);
}
function UserResourceInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'resourceType': json['resourceType'],
        'value': json['value'],
    };
}
function UserResourceInfoToJSON(json) {
    return UserResourceInfoToJSONTyped(json, false);
}
function UserResourceInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'resourceType': value['resourceType'],
        'value': value['value'],
    };
}
const propertyValidationAttributesMap = {};
function validateUserResourceInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
