// dana-node/dist/widget/v1/models/WidgetPaymentRequestAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWidgetPaymentRequestAdditionalInfo = instanceOfWidgetPaymentRequestAdditionalInfo;
exports.WidgetPaymentRequestAdditionalInfoFromJSON = WidgetPaymentRequestAdditionalInfoFromJSON;
exports.WidgetPaymentRequestAdditionalInfoFromJSONTyped = WidgetPaymentRequestAdditionalInfoFromJSONTyped;
exports.WidgetPaymentRequestAdditionalInfoToJSON = WidgetPaymentRequestAdditionalInfoToJSON;
exports.WidgetPaymentRequestAdditionalInfoToJSONTyped = WidgetPaymentRequestAdditionalInfoToJSONTyped;
exports.validateWidgetPaymentRequestAdditionalInfo = validateWidgetPaymentRequestAdditionalInfo;
const runtime_1 = require("../../../runtime");
const Order_1 = require("./Order");
const ServiceInfo_1 = require("./ServiceInfo");
const EnvInfo_1 = require("./EnvInfo");
/**
 * Check if a given object implements the WidgetPaymentRequestAdditionalInfo interface.
 */
function instanceOfWidgetPaymentRequestAdditionalInfo(value) {
    if (!('productCode' in value) || value['productCode'] === undefined)
        return false;
    if (!('order' in value) || value['order'] === undefined)
        return false;
    if (!('mcc' in value) || value['mcc'] === undefined)
        return false;
    if (!('envInfo' in value) || value['envInfo'] === undefined)
        return false;
    return true;
}
function WidgetPaymentRequestAdditionalInfoFromJSON(json) {
    return WidgetPaymentRequestAdditionalInfoFromJSONTyped(json, false);
}
function WidgetPaymentRequestAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'supportDeepLinkCheckoutUrl': json['supportDeepLinkCheckoutUrl'] == null ? undefined : json['supportDeepLinkCheckoutUrl'],
        'phoneNumber': json['phoneNumber'] == null ? undefined : json['phoneNumber'],
        'publicUserId': json['publicUserId'] == null ? undefined : json['publicUserId'],
        'productCode': json['productCode'],
        'serviceInfo': json['serviceInfo'] == null ? undefined : (0, ServiceInfo_1.ServiceInfoFromJSON)(json['serviceInfo']),
        'order': (0, Order_1.OrderFromJSON)(json['order']),
        'mcc': json['mcc'],
        'envInfo': (0, EnvInfo_1.EnvInfoFromJSON)(json['envInfo']),
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
    };
}
function WidgetPaymentRequestAdditionalInfoToJSON(json) {
    return WidgetPaymentRequestAdditionalInfoToJSONTyped(json, false);
}
function WidgetPaymentRequestAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'supportDeepLinkCheckoutUrl': value['supportDeepLinkCheckoutUrl'],
        'phoneNumber': value['phoneNumber'],
        'publicUserId': value['publicUserId'],
        'productCode': value['productCode'],
        'serviceInfo': (0, ServiceInfo_1.ServiceInfoToJSON)(value['serviceInfo']),
        'order': (0, Order_1.OrderToJSON)(value['order']),
        'mcc': value['mcc'],
        'envInfo': (0, EnvInfo_1.EnvInfoToJSON)(value['envInfo']),
        'extendInfo': value['extendInfo'],
    };
}
const propertyValidationAttributesMap = {
    supportDeepLinkCheckoutUrl: {
        maxLength: 64,
    },
    phoneNumber: {
        maxLength: 64,
    },
    publicUserId: {
        maxLength: 64,
    },
    productCode: {
        maxLength: 32,
    },
    mcc: {
        maxLength: 64,
    },
    extendInfo: {
        maxLength: 4096,
    },
};
function validateWidgetPaymentRequestAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('supportDeepLinkCheckoutUrl', value.supportDeepLinkCheckoutUrl, propertyValidationAttributesMap['supportDeepLinkCheckoutUrl']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('phoneNumber', value.phoneNumber, propertyValidationAttributesMap['phoneNumber']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('publicUserId', value.publicUserId, propertyValidationAttributesMap['publicUserId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('productCode', value.productCode, propertyValidationAttributesMap['productCode']));
    validationErrorContexts.push(...(0, ServiceInfo_1.validateServiceInfo)(value.serviceInfo));
    validationErrorContexts.push(...(0, Order_1.validateOrder)(value.order));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('mcc', value.mcc, propertyValidationAttributesMap['mcc']));
    validationErrorContexts.push(...(0, EnvInfo_1.validateEnvInfo)(value.envInfo));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    return validationErrorContexts;
}
