// dana-node/dist/widget/v1/models/WidgetPaymentResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWidgetPaymentResponse = instanceOfWidgetPaymentResponse;
exports.WidgetPaymentResponseFromJSON = WidgetPaymentResponseFromJSON;
exports.WidgetPaymentResponseFromJSONTyped = WidgetPaymentResponseFromJSONTyped;
exports.WidgetPaymentResponseToJSON = WidgetPaymentResponseToJSON;
exports.WidgetPaymentResponseToJSONTyped = WidgetPaymentResponseToJSONTyped;
exports.validateWidgetPaymentResponse = validateWidgetPaymentResponse;
const runtime_1 = require("../../../runtime");
/**
 * Check if a given object implements the WidgetPaymentResponse interface.
 */
function instanceOfWidgetPaymentResponse(value) {
    if (!('responseCode' in value) || value['responseCode'] === undefined)
        return false;
    if (!('responseMessage' in value) || value['responseMessage'] === undefined)
        return false;
    if (!('partnerReferenceNo' in value) || value['partnerReferenceNo'] === undefined)
        return false;
    return true;
}
function WidgetPaymentResponseFromJSON(json) {
    return WidgetPaymentResponseFromJSONTyped(json, false);
}
function WidgetPaymentResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'],
        'responseMessage': json['responseMessage'],
        'referenceNo': json['referenceNo'] == null ? undefined : json['referenceNo'],
        'partnerReferenceNo': json['partnerReferenceNo'],
        'webRedirectUrl': json['webRedirectUrl'] == null ? undefined : json['webRedirectUrl'],
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function WidgetPaymentResponseToJSON(json) {
    return WidgetPaymentResponseToJSONTyped(json, false);
}
function WidgetPaymentResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
        'referenceNo': value['referenceNo'],
        'partnerReferenceNo': value['partnerReferenceNo'],
        'webRedirectUrl': value['webRedirectUrl'],
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
    referenceNo: {
        maxLength: 64,
    },
    partnerReferenceNo: {
        maxLength: 64,
    },
    webRedirectUrl: {
        maxLength: 2048,
    },
};
function validateWidgetPaymentResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('referenceNo', value.referenceNo, propertyValidationAttributesMap['referenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('webRedirectUrl', value.webRedirectUrl, propertyValidationAttributesMap['webRedirectUrl']));
    return validationErrorContexts;
}
