// dana-node/dist/widget/v1/models/WidgetPaymentRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWidgetPaymentRequest = instanceOfWidgetPaymentRequest;
exports.WidgetPaymentRequestFromJSON = WidgetPaymentRequestFromJSON;
exports.WidgetPaymentRequestFromJSONTyped = WidgetPaymentRequestFromJSONTyped;
exports.WidgetPaymentRequestToJSON = WidgetPaymentRequestToJSON;
exports.WidgetPaymentRequestToJSONTyped = WidgetPaymentRequestToJSONTyped;
exports.validateWidgetPaymentRequest = validateWidgetPaymentRequest;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
const UrlParam_1 = require("./UrlParam");
const WidgetPaymentRequestAdditionalInfo_1 = require("./WidgetPaymentRequestAdditionalInfo");
const PayOptionDetail_1 = require("./PayOptionDetail");
/**
 * Check if a given object implements the WidgetPaymentRequest interface.
 */
function instanceOfWidgetPaymentRequest(value) {
    if (!('partnerReferenceNo' in value) || value['partnerReferenceNo'] === undefined)
        return false;
    if (!('merchantId' in value) || value['merchantId'] === undefined)
        return false;
    if (!('amount' in value) || value['amount'] === undefined)
        return false;
    if (!('additionalInfo' in value) || value['additionalInfo'] === undefined)
        return false;
    return true;
}
function WidgetPaymentRequestFromJSON(json) {
    return WidgetPaymentRequestFromJSONTyped(json, false);
}
function WidgetPaymentRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'partnerReferenceNo': json['partnerReferenceNo'],
        'merchantId': json['merchantId'],
        'subMerchantId': json['subMerchantId'] == null ? undefined : json['subMerchantId'],
        'amount': (0, Money_1.MoneyFromJSON)(json['amount']),
        'externalStoreId': json['externalStoreId'] == null ? undefined : json['externalStoreId'],
        'validUpTo': json['validUpTo'] == null ? undefined : json['validUpTo'],
        'pointOfInitiation': json['pointOfInitiation'] == null ? undefined : json['pointOfInitiation'],
        'disabledPayMethods': json['disabledPayMethods'] == null ? undefined : json['disabledPayMethods'],
        'payOptionDetails': json['payOptionDetails'] == null ? undefined : (json['payOptionDetails'].map(PayOptionDetail_1.PayOptionDetailFromJSON)),
        'additionalInfo': (0, WidgetPaymentRequestAdditionalInfo_1.WidgetPaymentRequestAdditionalInfoFromJSON)(json['additionalInfo']),
        'urlParams': json['urlParams'] == null ? undefined : (json['urlParams'].map(UrlParam_1.UrlParamFromJSON)),
    };
}
function WidgetPaymentRequestToJSON(json) {
    return WidgetPaymentRequestToJSONTyped(json, false);
}
function WidgetPaymentRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'partnerReferenceNo': value['partnerReferenceNo'],
        'merchantId': value['merchantId'],
        'subMerchantId': value['subMerchantId'],
        'amount': (0, Money_1.MoneyToJSON)(value['amount']),
        'externalStoreId': value['externalStoreId'],
        'validUpTo': value['validUpTo'],
        'pointOfInitiation': value['pointOfInitiation'],
        'disabledPayMethods': value['disabledPayMethods'],
        'payOptionDetails': value['payOptionDetails'] == null ? undefined : (value['payOptionDetails'].map(PayOptionDetail_1.PayOptionDetailToJSON)),
        'additionalInfo': (0, WidgetPaymentRequestAdditionalInfo_1.WidgetPaymentRequestAdditionalInfoToJSON)(value['additionalInfo']),
        'urlParams': value['urlParams'] == null ? undefined : (value['urlParams'].map(UrlParam_1.UrlParamToJSON)),
    };
}
const propertyValidationAttributesMap = {
    partnerReferenceNo: {
        maxLength: 64,
    },
    merchantId: {
        maxLength: 64,
    },
    subMerchantId: {
        maxLength: 32,
    },
    externalStoreId: {
        maxLength: 64,
    },
    validUpTo: {
        pattern: new RegExp('/^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+07:00$/'.slice(1, -1)),
    },
    pointOfInitiation: {
        maxLength: 20,
    },
    disabledPayMethods: {
        maxLength: 64,
    },
};
function validateWidgetPaymentRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantId', value.merchantId, propertyValidationAttributesMap['merchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('subMerchantId', value.subMerchantId, propertyValidationAttributesMap['subMerchantId']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.amount));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('externalStoreId', value.externalStoreId, propertyValidationAttributesMap['externalStoreId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('validUpTo', value.validUpTo, propertyValidationAttributesMap['validUpTo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('pointOfInitiation', value.pointOfInitiation, propertyValidationAttributesMap['pointOfInitiation']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('disabledPayMethods', value.disabledPayMethods, propertyValidationAttributesMap['disabledPayMethods']));
    validationErrorContexts.push(...(0, WidgetPaymentRequestAdditionalInfo_1.validateWidgetPaymentRequestAdditionalInfo)(value.additionalInfo));
    return validationErrorContexts;
}
