// dana-node/dist/payment_gateway/v1/models/ConsultPayResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfConsultPayResponse = instanceOfConsultPayResponse;
exports.ConsultPayResponseFromJSON = ConsultPayResponseFromJSON;
exports.ConsultPayResponseFromJSONTyped = ConsultPayResponseFromJSONTyped;
exports.ConsultPayResponseToJSON = ConsultPayResponseToJSON;
exports.ConsultPayResponseToJSONTyped = ConsultPayResponseToJSONTyped;
exports.validateConsultPayResponse = validateConsultPayResponse;
const runtime_1 = require("../../../runtime");
const ConsultPayPaymentInfo_1 = require("./ConsultPayPaymentInfo");
/**
 * Check if a given object implements the ConsultPayResponse interface.
 */
function instanceOfConsultPayResponse(value) {
    return true;
}
function ConsultPayResponseFromJSON(json) {
    return ConsultPayResponseFromJSONTyped(json, false);
}
function ConsultPayResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'] == null ? undefined : json['responseCode'],
        'responseMessage': json['responseMessage'] == null ? undefined : json['responseMessage'],
        'paymentInfos': json['paymentInfos'] == null ? undefined : (json['paymentInfos'].map(ConsultPayPaymentInfo_1.ConsultPayPaymentInfoFromJSON)),
    };
}
function ConsultPayResponseToJSON(json) {
    return ConsultPayResponseToJSONTyped(json, false);
}
function ConsultPayResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
        'paymentInfos': value['paymentInfos'] == null ? undefined : (value['paymentInfos'].map(ConsultPayPaymentInfo_1.ConsultPayPaymentInfoToJSON)),
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
};
function validateConsultPayResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    return validationErrorContexts;
}
