// dana-node/dist/payment_gateway/v1/models/ConsultPayRequestAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfConsultPayRequestAdditionalInfo = instanceOfConsultPayRequestAdditionalInfo;
exports.ConsultPayRequestAdditionalInfoFromJSON = ConsultPayRequestAdditionalInfoFromJSON;
exports.ConsultPayRequestAdditionalInfoFromJSONTyped = ConsultPayRequestAdditionalInfoFromJSONTyped;
exports.ConsultPayRequestAdditionalInfoToJSON = ConsultPayRequestAdditionalInfoToJSON;
exports.ConsultPayRequestAdditionalInfoToJSONTyped = ConsultPayRequestAdditionalInfoToJSONTyped;
exports.validateConsultPayRequestAdditionalInfo = validateConsultPayRequestAdditionalInfo;
const runtime_1 = require("../../../runtime");
const Buyer_1 = require("./Buyer");
const EnvInfo_1 = require("./EnvInfo");
/**
 * Check if a given object implements the ConsultPayRequestAdditionalInfo interface.
 */
function instanceOfConsultPayRequestAdditionalInfo(value) {
    if (!('buyer' in value) || value['buyer'] === undefined)
        return false;
    if (!('envInfo' in value) || value['envInfo'] === undefined)
        return false;
    return true;
}
function ConsultPayRequestAdditionalInfoFromJSON(json) {
    return ConsultPayRequestAdditionalInfoFromJSONTyped(json, false);
}
function ConsultPayRequestAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'buyer': (0, Buyer_1.BuyerFromJSON)(json['buyer']),
        'envInfo': (0, EnvInfo_1.EnvInfoFromJSON)(json['envInfo']),
        'merchantTransType': json['merchantTransType'] == null ? undefined : json['merchantTransType'],
    };
}
function ConsultPayRequestAdditionalInfoToJSON(json) {
    return ConsultPayRequestAdditionalInfoToJSONTyped(json, false);
}
function ConsultPayRequestAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'buyer': (0, Buyer_1.BuyerToJSON)(value['buyer']),
        'envInfo': (0, EnvInfo_1.EnvInfoToJSON)(value['envInfo']),
        'merchantTransType': value['merchantTransType'],
    };
}
const propertyValidationAttributesMap = {
    merchantTransType: {
        maxLength: 64,
    },
};
function validateConsultPayRequestAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, Buyer_1.validateBuyer)(value.buyer));
    validationErrorContexts.push(...(0, EnvInfo_1.validateEnvInfo)(value.envInfo));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantTransType', value.merchantTransType, propertyValidationAttributesMap['merchantTransType']));
    return validationErrorContexts;
}
