// dana-node/dist/payment_gateway/v1/models/CreateOrderByApiAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateOrderByApiAdditionalInfo = instanceOfCreateOrderByApiAdditionalInfo;
exports.CreateOrderByApiAdditionalInfoFromJSON = CreateOrderByApiAdditionalInfoFromJSON;
exports.CreateOrderByApiAdditionalInfoFromJSONTyped = CreateOrderByApiAdditionalInfoFromJSONTyped;
exports.CreateOrderByApiAdditionalInfoToJSON = CreateOrderByApiAdditionalInfoToJSON;
exports.CreateOrderByApiAdditionalInfoToJSONTyped = CreateOrderByApiAdditionalInfoToJSONTyped;
exports.validateCreateOrderByApiAdditionalInfo = validateCreateOrderByApiAdditionalInfo;
const runtime_1 = require("../../../runtime");
const OrderApiObject_1 = require("./OrderApiObject");
const EnvInfo_1 = require("./EnvInfo");
/**
 * Check if a given object implements the CreateOrderByApiAdditionalInfo interface.
 */
function instanceOfCreateOrderByApiAdditionalInfo(value) {
    if (!('mcc' in value) || value['mcc'] === undefined)
        return false;
    if (!('envInfo' in value) || value['envInfo'] === undefined)
        return false;
    return true;
}
function CreateOrderByApiAdditionalInfoFromJSON(json) {
    return CreateOrderByApiAdditionalInfoFromJSONTyped(json, false);
}
function CreateOrderByApiAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'mcc': json['mcc'],
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'envInfo': (0, EnvInfo_1.EnvInfoFromJSON)(json['envInfo']),
        'order': json['order'] == null ? undefined : (0, OrderApiObject_1.OrderApiObjectFromJSON)(json['order']),
    };
}
function CreateOrderByApiAdditionalInfoToJSON(json) {
    return CreateOrderByApiAdditionalInfoToJSONTyped(json, false);
}
function CreateOrderByApiAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'mcc': value['mcc'],
        'extendInfo': value['extendInfo'],
        'envInfo': (0, EnvInfo_1.EnvInfoToJSON)(value['envInfo']),
        'order': (0, OrderApiObject_1.OrderApiObjectToJSON)(value['order']),
    };
}
const propertyValidationAttributesMap = {
    mcc: {
        maxLength: 64,
    },
    extendInfo: {
        maxLength: 4096,
    },
};
function validateCreateOrderByApiAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('mcc', value.mcc, propertyValidationAttributesMap['mcc']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...(0, EnvInfo_1.validateEnvInfo)(value.envInfo));
    validationErrorContexts.push(...(0, OrderApiObject_1.validateOrderApiObject)(value.order));
    return validationErrorContexts;
}
