// dana-node/dist/payment_gateway/v1/models/CreateOrderRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrderRequestFromJSON = CreateOrderRequestFromJSON;
exports.CreateOrderRequestFromJSONTyped = CreateOrderRequestFromJSONTyped;
exports.CreateOrderRequestToJSON = CreateOrderRequestToJSON;
exports.CreateOrderRequestToJSONTyped = CreateOrderRequestToJSONTyped;
exports.validateCreateOrderRequest = validateCreateOrderRequest;
const CreateOrderByApiRequest_1 = require("./CreateOrderByApiRequest");
const CreateOrderByRedirectRequest_1 = require("./CreateOrderByRedirectRequest");
function CreateOrderRequestFromJSON(json) {
    return CreateOrderRequestFromJSONTyped(json, false);
}
function CreateOrderRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    if ((0, CreateOrderByApiRequest_1.instanceOfCreateOrderByApiRequest)(json)) {
        return (0, CreateOrderByApiRequest_1.CreateOrderByApiRequestFromJSONTyped)(json, true);
    }
    if ((0, CreateOrderByRedirectRequest_1.instanceOfCreateOrderByRedirectRequest)(json)) {
        return (0, CreateOrderByRedirectRequest_1.CreateOrderByRedirectRequestFromJSONTyped)(json, true);
    }
    return {};
}
function CreateOrderRequestToJSON(json) {
    return CreateOrderRequestToJSONTyped(json, false);
}
function CreateOrderRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    if ((0, CreateOrderByApiRequest_1.instanceOfCreateOrderByApiRequest)(value)) {
        return (0, CreateOrderByApiRequest_1.CreateOrderByApiRequestToJSON)(value);
    }
    if ((0, CreateOrderByRedirectRequest_1.instanceOfCreateOrderByRedirectRequest)(value)) {
        return (0, CreateOrderByRedirectRequest_1.CreateOrderByRedirectRequestToJSON)(value);
    }
    return {};
}
function validateCreateOrderRequest(value) {
    if (value == null) {
        return [];
    }
    if ((0, CreateOrderByApiRequest_1.instanceOfCreateOrderByApiRequest)(value)) {
        return (0, CreateOrderByApiRequest_1.validateCreateOrderByApiRequest)(value);
    }
    if ((0, CreateOrderByRedirectRequest_1.instanceOfCreateOrderByRedirectRequest)(value)) {
        return (0, CreateOrderByRedirectRequest_1.validateCreateOrderByRedirectRequest)(value);
    }
    return [];
}
