// dana-node/dist/payment_gateway/v1/models/CreateOrderResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateOrderResponse = instanceOfCreateOrderResponse;
exports.CreateOrderResponseFromJSON = CreateOrderResponseFromJSON;
exports.CreateOrderResponseFromJSONTyped = CreateOrderResponseFromJSONTyped;
exports.CreateOrderResponseToJSON = CreateOrderResponseToJSON;
exports.CreateOrderResponseToJSONTyped = CreateOrderResponseToJSONTyped;
exports.validateCreateOrderResponse = validateCreateOrderResponse;
const runtime_1 = require("../../../runtime");
const CreateOrderResponseAdditionalInfo_1 = require("./CreateOrderResponseAdditionalInfo");
/**
 * Check if a given object implements the CreateOrderResponse interface.
 */
function instanceOfCreateOrderResponse(value) {
    if (!('responseCode' in value) || value['responseCode'] === undefined)
        return false;
    if (!('responseMessage' in value) || value['responseMessage'] === undefined)
        return false;
    if (!('partnerReferenceNo' in value) || value['partnerReferenceNo'] === undefined)
        return false;
    return true;
}
function CreateOrderResponseFromJSON(json) {
    return CreateOrderResponseFromJSONTyped(json, false);
}
function CreateOrderResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'],
        'responseMessage': json['responseMessage'],
        'referenceNo': json['referenceNo'] == null ? undefined : json['referenceNo'],
        'partnerReferenceNo': json['partnerReferenceNo'],
        'webRedirectUrl': json['webRedirectUrl'] == null ? undefined : json['webRedirectUrl'],
        'additionalInfo': json['additionalInfo'] == null ? undefined : (0, CreateOrderResponseAdditionalInfo_1.CreateOrderResponseAdditionalInfoFromJSON)(json['additionalInfo']),
        'externalOrderId': json['externalOrderId'] == null ? undefined : json['externalOrderId'],
    };
}
function CreateOrderResponseToJSON(json) {
    return CreateOrderResponseToJSONTyped(json, false);
}
function CreateOrderResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
        'referenceNo': value['referenceNo'],
        'partnerReferenceNo': value['partnerReferenceNo'],
        'webRedirectUrl': value['webRedirectUrl'],
        'additionalInfo': (0, CreateOrderResponseAdditionalInfo_1.CreateOrderResponseAdditionalInfoToJSON)(value['additionalInfo']),
        'externalOrderId': value['externalOrderId'],
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
    referenceNo: {
        maxLength: 64,
    },
    partnerReferenceNo: {
        maxLength: 64,
    },
    webRedirectUrl: {
        maxLength: 2048,
    },
};
function validateCreateOrderResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('referenceNo', value.referenceNo, propertyValidationAttributesMap['referenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerReferenceNo', value.partnerReferenceNo, propertyValidationAttributesMap['partnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('webRedirectUrl', value.webRedirectUrl, propertyValidationAttributesMap['webRedirectUrl']));
    validationErrorContexts.push(...(0, CreateOrderResponseAdditionalInfo_1.validateCreateOrderResponseAdditionalInfo)(value.additionalInfo));
    return validationErrorContexts;
}
