// dana-node/dist/payment_gateway/v1/models/CreateOrderResponseAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateOrderResponseAdditionalInfo = instanceOfCreateOrderResponseAdditionalInfo;
exports.CreateOrderResponseAdditionalInfoFromJSON = CreateOrderResponseAdditionalInfoFromJSON;
exports.CreateOrderResponseAdditionalInfoFromJSONTyped = CreateOrderResponseAdditionalInfoFromJSONTyped;
exports.CreateOrderResponseAdditionalInfoToJSON = CreateOrderResponseAdditionalInfoToJSON;
exports.CreateOrderResponseAdditionalInfoToJSONTyped = CreateOrderResponseAdditionalInfoToJSONTyped;
exports.validateCreateOrderResponseAdditionalInfo = validateCreateOrderResponseAdditionalInfo;
/**
 * Check if a given object implements the CreateOrderResponseAdditionalInfo interface.
 */
function instanceOfCreateOrderResponseAdditionalInfo(value) {
    return true;
}
function CreateOrderResponseAdditionalInfoFromJSON(json) {
    return CreateOrderResponseAdditionalInfoFromJSONTyped(json, false);
}
function CreateOrderResponseAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'paymentCode': json['paymentCode'] == null ? undefined : json['paymentCode'],
    };
}
function CreateOrderResponseAdditionalInfoToJSON(json) {
    return CreateOrderResponseAdditionalInfoToJSONTyped(json, false);
}
function CreateOrderResponseAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'paymentCode': value['paymentCode'],
    };
}
const propertyValidationAttributesMap = {};
function validateCreateOrderResponseAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    return validationErrorContexts;
}
