// dana-node/dist/payment_gateway/v1/models/CreateOrderByRedirectAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateOrderByRedirectAdditionalInfo = instanceOfCreateOrderByRedirectAdditionalInfo;
exports.CreateOrderByRedirectAdditionalInfoFromJSON = CreateOrderByRedirectAdditionalInfoFromJSON;
exports.CreateOrderByRedirectAdditionalInfoFromJSONTyped = CreateOrderByRedirectAdditionalInfoFromJSONTyped;
exports.CreateOrderByRedirectAdditionalInfoToJSON = CreateOrderByRedirectAdditionalInfoToJSON;
exports.CreateOrderByRedirectAdditionalInfoToJSONTyped = CreateOrderByRedirectAdditionalInfoToJSONTyped;
exports.validateCreateOrderByRedirectAdditionalInfo = validateCreateOrderByRedirectAdditionalInfo;
const runtime_1 = require("../../../runtime");
const EnvInfo_1 = require("./EnvInfo");
const OrderRedirectObject_1 = require("./OrderRedirectObject");
/**
 * Check if a given object implements the CreateOrderByRedirectAdditionalInfo interface.
 */
function instanceOfCreateOrderByRedirectAdditionalInfo(value) {
    if (!('mcc' in value) || value['mcc'] === undefined)
        return false;
    if (!('envInfo' in value) || value['envInfo'] === undefined)
        return false;
    return true;
}
function CreateOrderByRedirectAdditionalInfoFromJSON(json) {
    return CreateOrderByRedirectAdditionalInfoFromJSONTyped(json, false);
}
function CreateOrderByRedirectAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'mcc': json['mcc'],
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'envInfo': (0, EnvInfo_1.EnvInfoFromJSON)(json['envInfo']),
        'order': json['order'] == null ? undefined : (0, OrderRedirectObject_1.OrderRedirectObjectFromJSON)(json['order']),
    };
}
function CreateOrderByRedirectAdditionalInfoToJSON(json) {
    return CreateOrderByRedirectAdditionalInfoToJSONTyped(json, false);
}
function CreateOrderByRedirectAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'mcc': value['mcc'],
        'extendInfo': value['extendInfo'],
        'envInfo': (0, EnvInfo_1.EnvInfoToJSON)(value['envInfo']),
        'order': (0, OrderRedirectObject_1.OrderRedirectObjectToJSON)(value['order']),
    };
}
const propertyValidationAttributesMap = {
    mcc: {
        maxLength: 64,
    },
    extendInfo: {
        maxLength: 4096,
    },
};
function validateCreateOrderByRedirectAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('mcc', value.mcc, propertyValidationAttributesMap['mcc']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...(0, EnvInfo_1.validateEnvInfo)(value.envInfo));
    validationErrorContexts.push(...(0, OrderRedirectObject_1.validateOrderRedirectObject)(value.order));
    return validationErrorContexts;
}
