// dana-node/dist/payment_gateway/v1/models/CreateOrderBaseAdditionalInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateOrderBaseAdditionalInfo = instanceOfCreateOrderBaseAdditionalInfo;
exports.CreateOrderBaseAdditionalInfoFromJSON = CreateOrderBaseAdditionalInfoFromJSON;
exports.CreateOrderBaseAdditionalInfoFromJSONTyped = CreateOrderBaseAdditionalInfoFromJSONTyped;
exports.CreateOrderBaseAdditionalInfoToJSON = CreateOrderBaseAdditionalInfoToJSON;
exports.CreateOrderBaseAdditionalInfoToJSONTyped = CreateOrderBaseAdditionalInfoToJSONTyped;
exports.validateCreateOrderBaseAdditionalInfo = validateCreateOrderBaseAdditionalInfo;
const runtime_1 = require("../../../runtime");
const EnvInfo_1 = require("./EnvInfo");
/**
 * Check if a given object implements the CreateOrderBaseAdditionalInfo interface.
 */
function instanceOfCreateOrderBaseAdditionalInfo(value) {
    if (!('mcc' in value) || value['mcc'] === undefined)
        return false;
    if (!('envInfo' in value) || value['envInfo'] === undefined)
        return false;
    return true;
}
function CreateOrderBaseAdditionalInfoFromJSON(json) {
    return CreateOrderBaseAdditionalInfoFromJSONTyped(json, false);
}
function CreateOrderBaseAdditionalInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'mcc': json['mcc'],
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'envInfo': (0, EnvInfo_1.EnvInfoFromJSON)(json['envInfo']),
    };
}
function CreateOrderBaseAdditionalInfoToJSON(json) {
    return CreateOrderBaseAdditionalInfoToJSONTyped(json, false);
}
function CreateOrderBaseAdditionalInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'mcc': value['mcc'],
        'extendInfo': value['extendInfo'],
        'envInfo': (0, EnvInfo_1.EnvInfoToJSON)(value['envInfo']),
    };
}
const propertyValidationAttributesMap = {
    mcc: {
        maxLength: 64,
    },
    extendInfo: {
        maxLength: 4096,
    },
};
function validateCreateOrderBaseAdditionalInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('mcc', value.mcc, propertyValidationAttributesMap['mcc']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...(0, EnvInfo_1.validateEnvInfo)(value.envInfo));
    return validationErrorContexts;
}
