// dana-node/dist/payment_gateway/v1/models/EnvInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnvInfoOrderTerminalTypeEnum = exports.EnvInfoTerminalTypeEnum = exports.EnvInfoSourcePlatformEnum = void 0;
exports.instanceOfEnvInfo = instanceOfEnvInfo;
exports.EnvInfoFromJSON = EnvInfoFromJSON;
exports.EnvInfoFromJSONTyped = EnvInfoFromJSONTyped;
exports.EnvInfoToJSON = EnvInfoToJSON;
exports.EnvInfoToJSONTyped = EnvInfoToJSONTyped;
exports.validateEnvInfo = validateEnvInfo;
const runtime_1 = require("../../../runtime");
/**
 * @export
 */
exports.EnvInfoSourcePlatformEnum = {
    Ipg: 'IPG'
};
/**
 * @export
 */
exports.EnvInfoTerminalTypeEnum = {
    App: 'APP',
    Web: 'WEB',
    Wap: 'WAP',
    System: 'SYSTEM'
};
/**
 * @export
 */
exports.EnvInfoOrderTerminalTypeEnum = {
    App: 'APP',
    Web: 'WEB',
    Wap: 'WAP',
    System: 'SYSTEM'
};
/**
 * Check if a given object implements the EnvInfo interface.
 */
function instanceOfEnvInfo(value) {
    if (!('sourcePlatform' in value) || value['sourcePlatform'] === undefined)
        return false;
    if (!('terminalType' in value) || value['terminalType'] === undefined)
        return false;
    return true;
}
function EnvInfoFromJSON(json) {
    return EnvInfoFromJSONTyped(json, false);
}
function EnvInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'sessionId': json['sessionId'] == null ? undefined : json['sessionId'],
        'tokenId': json['tokenId'] == null ? undefined : json['tokenId'],
        'websiteLanguage': json['websiteLanguage'] == null ? undefined : json['websiteLanguage'],
        'clientIp': json['clientIp'] == null ? undefined : json['clientIp'],
        'osType': json['osType'] == null ? undefined : json['osType'],
        'appVersion': json['appVersion'] == null ? undefined : json['appVersion'],
        'sdkVersion': json['sdkVersion'] == null ? undefined : json['sdkVersion'],
        'sourcePlatform': json['sourcePlatform'],
        'orderOsType': json['orderOsType'] == null ? undefined : json['orderOsType'],
        'merchantAppVersion': json['merchantAppVersion'] == null ? undefined : json['merchantAppVersion'],
        'terminalType': json['terminalType'],
        'orderTerminalType': json['orderTerminalType'] == null ? undefined : json['orderTerminalType'],
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'clientKey': json['clientKey'] == null ? undefined : json['clientKey'],
    };
}
function EnvInfoToJSON(json) {
    return EnvInfoToJSONTyped(json, false);
}
function EnvInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'sessionId': value['sessionId'],
        'tokenId': value['tokenId'],
        'websiteLanguage': value['websiteLanguage'],
        'clientIp': value['clientIp'],
        'osType': value['osType'],
        'appVersion': value['appVersion'],
        'sdkVersion': value['sdkVersion'],
        'sourcePlatform': value['sourcePlatform'],
        'orderOsType': value['orderOsType'],
        'merchantAppVersion': value['merchantAppVersion'],
        'terminalType': value['terminalType'],
        'orderTerminalType': value['orderTerminalType'],
        'extendInfo': value['extendInfo'],
        'clientKey': value['clientKey'],
    };
}
const propertyValidationAttributesMap = {
    sessionId: {
        maxLength: 128,
    },
    tokenId: {
        maxLength: 128,
    },
    websiteLanguage: {
        maxLength: 16,
    },
    clientIp: {
        maxLength: 32,
    },
    osType: {
        maxLength: 128,
    },
    appVersion: {
        maxLength: 128,
    },
    sdkVersion: {
        maxLength: 128,
    },
    sourcePlatform: {
        maxLength: 32,
    },
    orderOsType: {
        maxLength: 128,
    },
    merchantAppVersion: {
        maxLength: 128,
    },
    terminalType: {
        maxLength: 32,
    },
    orderTerminalType: {
        maxLength: 32,
    },
    extendInfo: {
        maxLength: 4096,
    },
    clientKey: {
        maxLength: 64,
    },
};
function validateEnvInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('sessionId', value.sessionId, propertyValidationAttributesMap['sessionId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('tokenId', value.tokenId, propertyValidationAttributesMap['tokenId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('websiteLanguage', value.websiteLanguage, propertyValidationAttributesMap['websiteLanguage']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientIp', value.clientIp, propertyValidationAttributesMap['clientIp']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('osType', value.osType, propertyValidationAttributesMap['osType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('appVersion', value.appVersion, propertyValidationAttributesMap['appVersion']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('sdkVersion', value.sdkVersion, propertyValidationAttributesMap['sdkVersion']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('sourcePlatform', value.sourcePlatform, propertyValidationAttributesMap['sourcePlatform']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('orderOsType', value.orderOsType, propertyValidationAttributesMap['orderOsType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantAppVersion', value.merchantAppVersion, propertyValidationAttributesMap['merchantAppVersion']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('terminalType', value.terminalType, propertyValidationAttributesMap['terminalType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('orderTerminalType', value.orderTerminalType, propertyValidationAttributesMap['orderTerminalType']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('clientKey', value.clientKey, propertyValidationAttributesMap['clientKey']));
    return validationErrorContexts;
}
