// dana-node/dist/payment_gateway/v1/models/OrderRedirectObject.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderRedirectObjectScenarioEnum = void 0;
exports.instanceOfOrderRedirectObject = instanceOfOrderRedirectObject;
exports.OrderRedirectObjectFromJSON = OrderRedirectObjectFromJSON;
exports.OrderRedirectObjectFromJSONTyped = OrderRedirectObjectFromJSONTyped;
exports.OrderRedirectObjectToJSON = OrderRedirectObjectToJSON;
exports.OrderRedirectObjectToJSONTyped = OrderRedirectObjectToJSONTyped;
exports.validateOrderRedirectObject = validateOrderRedirectObject;
const runtime_1 = require("../../../runtime");
const Buyer_1 = require("./Buyer");
const Goods_1 = require("./Goods");
const ShippingInfo_1 = require("./ShippingInfo");
/**
 * @export
 */
exports.OrderRedirectObjectScenarioEnum = {
    Redirect: 'REDIRECT'
};
/**
 * Check if a given object implements the OrderRedirectObject interface.
 */
function instanceOfOrderRedirectObject(value) {
    if (!('orderTitle' in value) || value['orderTitle'] === undefined)
        return false;
    return true;
}
function OrderRedirectObjectFromJSON(json) {
    return OrderRedirectObjectFromJSONTyped(json, false);
}
function OrderRedirectObjectFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'orderTitle': json['orderTitle'],
        'merchantTransType': json['merchantTransType'] == null ? undefined : json['merchantTransType'],
        'buyer': json['buyer'] == null ? undefined : (0, Buyer_1.BuyerFromJSON)(json['buyer']),
        'goods': json['goods'] == null ? undefined : (json['goods'].map(Goods_1.GoodsFromJSON)),
        'shippingInfo': json['shippingInfo'] == null ? undefined : (json['shippingInfo'].map(ShippingInfo_1.ShippingInfoFromJSON)),
        'extendInfo': json['extendInfo'] == null ? undefined : json['extendInfo'],
        'createdTime': json['createdTime'] == null ? undefined : json['createdTime'],
        'orderMemo': json['orderMemo'] == null ? undefined : json['orderMemo'],
        'scenario': json['scenario'] == null ? undefined : json['scenario'],
    };
}
function OrderRedirectObjectToJSON(json) {
    return OrderRedirectObjectToJSONTyped(json, false);
}
function OrderRedirectObjectToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'orderTitle': value['orderTitle'],
        'merchantTransType': value['merchantTransType'],
        'buyer': (0, Buyer_1.BuyerToJSON)(value['buyer']),
        'goods': value['goods'] == null ? undefined : (value['goods'].map(Goods_1.GoodsToJSON)),
        'shippingInfo': value['shippingInfo'] == null ? undefined : (value['shippingInfo'].map(ShippingInfo_1.ShippingInfoToJSON)),
        'extendInfo': value['extendInfo'],
        'createdTime': value['createdTime'],
        'orderMemo': value['orderMemo'],
        'scenario': value['scenario'],
    };
}
const propertyValidationAttributesMap = {
    orderTitle: {
        maxLength: 64,
    },
    merchantTransType: {
        maxLength: 64,
    },
    extendInfo: {
        maxLength: 4096,
    },
    scenario: {
        maxLength: 64,
    },
};
function validateOrderRedirectObject(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('orderTitle', value.orderTitle, propertyValidationAttributesMap['orderTitle']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantTransType', value.merchantTransType, propertyValidationAttributesMap['merchantTransType']));
    validationErrorContexts.push(...(0, Buyer_1.validateBuyer)(value.buyer));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('extendInfo', value.extendInfo, propertyValidationAttributesMap['extendInfo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('scenario', value.scenario, propertyValidationAttributesMap['scenario']));
    return validationErrorContexts;
}
