// dana-node/dist/payment_gateway/v1/models/PromoInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPromoInfo = instanceOfPromoInfo;
exports.PromoInfoFromJSON = PromoInfoFromJSON;
exports.PromoInfoFromJSONTyped = PromoInfoFromJSONTyped;
exports.PromoInfoToJSON = PromoInfoToJSON;
exports.PromoInfoToJSONTyped = PromoInfoToJSONTyped;
exports.validatePromoInfo = validatePromoInfo;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * Check if a given object implements the PromoInfo interface.
 */
function instanceOfPromoInfo(value) {
    if (!('promoAmount' in value) || value['promoAmount'] === undefined)
        return false;
    if (!('promoId' in value) || value['promoId'] === undefined)
        return false;
    if (!('promoType' in value) || value['promoType'] === undefined)
        return false;
    return true;
}
function PromoInfoFromJSON(json) {
    return PromoInfoFromJSONTyped(json, false);
}
function PromoInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'promoAmount': (0, Money_1.MoneyFromJSON)(json['promoAmount']),
        'promoId': json['promoId'],
        'promoType': json['promoType'],
    };
}
function PromoInfoToJSON(json) {
    return PromoInfoToJSONTyped(json, false);
}
function PromoInfoToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'promoAmount': (0, Money_1.MoneyToJSON)(value['promoAmount']),
        'promoId': value['promoId'],
        'promoType': value['promoType'],
    };
}
const propertyValidationAttributesMap = {
    promoId: {
        maxLength: 64,
    },
    promoType: {
        maxLength: 32,
    },
};
function validatePromoInfo(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.promoAmount));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('promoId', value.promoId, propertyValidationAttributesMap['promoId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('promoType', value.promoType, propertyValidationAttributesMap['promoType']));
    return validationErrorContexts;
}
