// dana-node/dist/payment_gateway/v1/models/QueryPaymentRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfQueryPaymentRequest = instanceOfQueryPaymentRequest;
exports.QueryPaymentRequestFromJSON = QueryPaymentRequestFromJSON;
exports.QueryPaymentRequestFromJSONTyped = QueryPaymentRequestFromJSONTyped;
exports.QueryPaymentRequestToJSON = QueryPaymentRequestToJSON;
exports.QueryPaymentRequestToJSONTyped = QueryPaymentRequestToJSONTyped;
exports.validateQueryPaymentRequest = validateQueryPaymentRequest;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * Check if a given object implements the QueryPaymentRequest interface.
 */
function instanceOfQueryPaymentRequest(value) {
    if (!('serviceCode' in value) || value['serviceCode'] === undefined)
        return false;
    if (!('merchantId' in value) || value['merchantId'] === undefined)
        return false;
    return true;
}
function QueryPaymentRequestFromJSON(json) {
    return QueryPaymentRequestFromJSONTyped(json, false);
}
function QueryPaymentRequestFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'originalPartnerReferenceNo': json['originalPartnerReferenceNo'] == null ? undefined : json['originalPartnerReferenceNo'],
        'originalReferenceNo': json['originalReferenceNo'] == null ? undefined : json['originalReferenceNo'],
        'originalExternalId': json['originalExternalId'] == null ? undefined : json['originalExternalId'],
        'serviceCode': json['serviceCode'],
        'transactionDate': json['transactionDate'] == null ? undefined : json['transactionDate'],
        'amount': json['amount'] == null ? undefined : (0, Money_1.MoneyFromJSON)(json['amount']),
        'merchantId': json['merchantId'],
        'subMerchantId': json['subMerchantId'] == null ? undefined : json['subMerchantId'],
        'externalStoreId': json['externalStoreId'] == null ? undefined : json['externalStoreId'],
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function QueryPaymentRequestToJSON(json) {
    return QueryPaymentRequestToJSONTyped(json, false);
}
function QueryPaymentRequestToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'originalPartnerReferenceNo': value['originalPartnerReferenceNo'],
        'originalReferenceNo': value['originalReferenceNo'],
        'originalExternalId': value['originalExternalId'],
        'serviceCode': value['serviceCode'],
        'transactionDate': value['transactionDate'],
        'amount': (0, Money_1.MoneyToJSON)(value['amount']),
        'merchantId': value['merchantId'],
        'subMerchantId': value['subMerchantId'],
        'externalStoreId': value['externalStoreId'],
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    originalPartnerReferenceNo: {
        maxLength: 64,
    },
    originalReferenceNo: {
        maxLength: 64,
    },
    originalExternalId: {
        maxLength: 36,
    },
    serviceCode: {
        maxLength: 2,
    },
    transactionDate: {
        maxLength: 25,
        pattern: new RegExp('/^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+07:00$/'.slice(1, -1)),
    },
    merchantId: {
        maxLength: 64,
    },
    subMerchantId: {
        maxLength: 32,
    },
    externalStoreId: {
        maxLength: 64,
    },
};
function validateQueryPaymentRequest(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalPartnerReferenceNo', value.originalPartnerReferenceNo, propertyValidationAttributesMap['originalPartnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalReferenceNo', value.originalReferenceNo, propertyValidationAttributesMap['originalReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalExternalId', value.originalExternalId, propertyValidationAttributesMap['originalExternalId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('serviceCode', value.serviceCode, propertyValidationAttributesMap['serviceCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('transactionDate', value.transactionDate, propertyValidationAttributesMap['transactionDate']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.amount));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('merchantId', value.merchantId, propertyValidationAttributesMap['merchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('subMerchantId', value.subMerchantId, propertyValidationAttributesMap['subMerchantId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('externalStoreId', value.externalStoreId, propertyValidationAttributesMap['externalStoreId']));
    return validationErrorContexts;
}
