// dana-node/dist/payment_gateway/v1/models/RefundOrderResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * NOTE: This class is auto generated by DANA Indonesia.
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfRefundOrderResponse = instanceOfRefundOrderResponse;
exports.RefundOrderResponseFromJSON = RefundOrderResponseFromJSON;
exports.RefundOrderResponseFromJSONTyped = RefundOrderResponseFromJSONTyped;
exports.RefundOrderResponseToJSON = RefundOrderResponseToJSON;
exports.RefundOrderResponseToJSONTyped = RefundOrderResponseToJSONTyped;
exports.validateRefundOrderResponse = validateRefundOrderResponse;
const runtime_1 = require("../../../runtime");
const Money_1 = require("./Money");
/**
 * Check if a given object implements the RefundOrderResponse interface.
 */
function instanceOfRefundOrderResponse(value) {
    if (!('responseCode' in value) || value['responseCode'] === undefined)
        return false;
    if (!('responseMessage' in value) || value['responseMessage'] === undefined)
        return false;
    if (!('originalPartnerReferenceNo' in value) || value['originalPartnerReferenceNo'] === undefined)
        return false;
    if (!('partnerRefundNo' in value) || value['partnerRefundNo'] === undefined)
        return false;
    if (!('refundAmount' in value) || value['refundAmount'] === undefined)
        return false;
    return true;
}
function RefundOrderResponseFromJSON(json) {
    return RefundOrderResponseFromJSONTyped(json, false);
}
function RefundOrderResponseFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'responseCode': json['responseCode'],
        'responseMessage': json['responseMessage'],
        'originalReferenceNo': json['originalReferenceNo'] == null ? undefined : json['originalReferenceNo'],
        'originalPartnerReferenceNo': json['originalPartnerReferenceNo'],
        'originalExternalId': json['originalExternalId'] == null ? undefined : json['originalExternalId'],
        'originalCaptureNo': json['originalCaptureNo'] == null ? undefined : json['originalCaptureNo'],
        'refundNo': json['refundNo'] == null ? undefined : json['refundNo'],
        'partnerRefundNo': json['partnerRefundNo'],
        'refundAmount': (0, Money_1.MoneyFromJSON)(json['refundAmount']),
        'refundTime': json['refundTime'] == null ? undefined : json['refundTime'],
        'additionalInfo': json['additionalInfo'] == null ? undefined : json['additionalInfo'],
    };
}
function RefundOrderResponseToJSON(json) {
    return RefundOrderResponseToJSONTyped(json, false);
}
function RefundOrderResponseToJSONTyped(value, ignoreDiscriminator = false) {
    if (value == null) {
        return value;
    }
    return {
        'responseCode': value['responseCode'],
        'responseMessage': value['responseMessage'],
        'originalReferenceNo': value['originalReferenceNo'],
        'originalPartnerReferenceNo': value['originalPartnerReferenceNo'],
        'originalExternalId': value['originalExternalId'],
        'originalCaptureNo': value['originalCaptureNo'],
        'refundNo': value['refundNo'],
        'partnerRefundNo': value['partnerRefundNo'],
        'refundAmount': (0, Money_1.MoneyToJSON)(value['refundAmount']),
        'refundTime': value['refundTime'],
        'additionalInfo': value['additionalInfo'],
    };
}
const propertyValidationAttributesMap = {
    responseCode: {
        maxLength: 7,
    },
    responseMessage: {
        maxLength: 150,
    },
    originalReferenceNo: {
        maxLength: 64,
    },
    originalPartnerReferenceNo: {
        maxLength: 64,
    },
    originalExternalId: {
        maxLength: 36,
    },
    originalCaptureNo: {
        maxLength: 64,
    },
    refundNo: {
        maxLength: 64,
    },
    partnerRefundNo: {
        maxLength: 64,
    },
    refundTime: {
        maxLength: 25,
        pattern: new RegExp('/^\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}\\+07:00$/'.slice(1, -1)),
    },
};
function validateRefundOrderResponse(value) {
    const validationErrorContexts = [];
    if (value == null) {
        return validationErrorContexts;
    }
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseCode', value.responseCode, propertyValidationAttributesMap['responseCode']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('responseMessage', value.responseMessage, propertyValidationAttributesMap['responseMessage']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalReferenceNo', value.originalReferenceNo, propertyValidationAttributesMap['originalReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalPartnerReferenceNo', value.originalPartnerReferenceNo, propertyValidationAttributesMap['originalPartnerReferenceNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalExternalId', value.originalExternalId, propertyValidationAttributesMap['originalExternalId']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('originalCaptureNo', value.originalCaptureNo, propertyValidationAttributesMap['originalCaptureNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('refundNo', value.refundNo, propertyValidationAttributesMap['refundNo']));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('partnerRefundNo', value.partnerRefundNo, propertyValidationAttributesMap['partnerRefundNo']));
    validationErrorContexts.push(...(0, Money_1.validateMoney)(value.refundAmount));
    validationErrorContexts.push(...runtime_1.ValidationUtil.validateProperty('refundTime', value.refundTime, propertyValidationAttributesMap['refundTime']));
    return validationErrorContexts;
}
